<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="This R package supports the integration of scATAC-seq and scRNA-seq data using which one can determine
    cis-regulatory elements linked to gene to define domains of regulatory chromatin (DORCs), and TF-gene associations using FigR.
    This framework is associated with (our preprint)[https://www.biorxiv.org/content/10.1101/2021.07.28.453784v1]">
<title>Functional Inference of Gene Regulation (FigR) using paired single cell ATAC-seq and RNA-seq data â€¢ FigR</title>
<script src="deps/jquery-3.5.1/jquery-3.5.1.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Functional Inference of Gene Regulation (FigR) using paired single cell ATAC-seq and RNA-seq data">
<meta property="og:description" content="This R package supports the integration of scATAC-seq and scRNA-seq data using which one can determine
    cis-regulatory elements linked to gene to define domains of regulatory chromatin (DORCs), and TF-gene associations using FigR.
    This framework is associated with (our preprint)[https://www.biorxiv.org/content/10.1101/2021.07.28.453784v1]">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">FigR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-tutorials">Tutorials</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-tutorials">
    <a class="dropdown-item" href="articles/FigR_stim.html">Pairing scATAC and scRNA-seq</a>
    <a class="dropdown-item" href="articles/FigR_shareseq.html">FigR on multi-modal data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">
    <span class="fa fa-gears"></span>
     
    Functions
  </a>
</li>
<h6 class="dropdown-header" data-toc-skip>News</h6>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/buenrostrolab/FigR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="functional-inference-of-gene-regulation-figr">Functional Inference of Gene Regulation (FigR)<a class="anchor" aria-label="anchor" href="#functional-inference-of-gene-regulation-figr"></a>
</h1></div>
<p>FigR is a computational framework for supporting the integration of single cell chromatin accessibility and gene expression data to infer transcriptional regulators of target genes</p>
<p align="center">
<img src="reference/figures/FigR.PNG" height="50%" width="50%"></p>
<p>FigR uses independently or concomitantly-profiled single-cell ATAC-seq (scATAC-seq) and scRNA-seq, to i) computationally-pair scATAC-seq and scRNA-seq datasets (if needed), ii) infers cis-regulatory interactions, and iii) defines a TF-gene gene regulatory network (GRN)</p>
</div>
<div class="section level1">
<h1 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h1>
<p><code>devtools::install_github("buenrostrolab/FigR")</code></p>
</div>
<div class="section level1">
<h1 id="quickstart">Quickstart<a class="anchor" aria-label="anchor" href="#quickstart"></a>
</h1>
<div class="section level3">
<h3 id="dorc-calling">DORC calling<a class="anchor" aria-label="anchor" href="#dorc-calling"></a>
</h3>
<p>If you have a <code>SummarizedExperiment</code> object of the scATAC-seq reads in peaks counts (peaks x cells), as well as a <code>sparseMatrix</code> object of the scRNA-seq gene counts (genes x cells) that are paired (either computationally or true multi-modal), you can determine <em>cis</em>-regulatory associations (gene-peak correlations) as follows</p>
<pre><code><span class="co"># Run using multiple cores if parallel support</span>
<span class="va">cisCor</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/runGenePeakcorr.html">runGenePeakcorr</a></span><span class="op">(</span>ATAC.se <span class="op">=</span> <span class="va">ATAC.SE</span>,
                           RNAmat <span class="op">=</span> <span class="va">rnaMat</span>,
                           genome <span class="op">=</span> <span class="st">"hg19"</span>, <span class="co"># Also supports mm10 and hg38</span>
                           nCores <span class="op">=</span> <span class="fl">4</span>, 
                           p.cut<span class="op">=</span><span class="cn">NULL</span><span class="op">)</span>
                    
<span class="co"># Filter peak-gene correlations by p-value                    </span>
<span class="va">cisCor.filt</span> <span class="op">&lt;-</span> <span class="va">cisCor</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pvalZ</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span>

<span class="co"># Determine DORC genes</span>
<span class="va">dorcGenes</span> <span class="op">&lt;-</span> <span class="va">cisCor.filt</span> <span class="op">%&gt;%</span> <span class="fu">dorcJplot</span><span class="op">(</span><span class="op">)</span>

<span class="co"># Get DORC scores</span>
<span class="va">dorcMat</span> <span class="op">&lt;-</span> <span class="fu">getDORCscores</span><span class="op">(</span><span class="va">ATAC.SE</span>,dorcTab<span class="op">=</span><span class="va">cisCor.filt</span>,geneList<span class="op">=</span><span class="va">dorcGenes</span>,nCores<span class="op">=</span><span class="fl">4</span><span class="op">)</span>

<span class="co"># Smooth DORC scores (using cell KNNs)</span>
<span class="va">dorcMat.smooth</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/smoothScoresNN.html">smoothScoresNN</a></span><span class="op">(</span>NNmat<span class="op">=</span><span class="va">cellKNN.mat</span>,mat<span class="op">=</span><span class="va">dorcMat</span>,nCores<span class="op">=</span><span class="fl">4</span><span class="op">)</span></code></pre>
<p><em>Cis</em>-regulatory correlation analysis framework to identify gene-peak (chromatin accessibility peak) significant associations and deduce key genes that are domains of such regulatory activity (DORCs)</p>
</div>
<div class="section level3">
<h3 id="figr">FigR<a class="anchor" aria-label="anchor" href="#figr"></a>
</h3>
<p>Once you have your DORC definitions, you can use this, along with the paired RNA data, to run the core FigR function. This returns a table of regulations cores between each DORC gene and a TF in the reference database, which we can filter on and visualize (see section below)</p>
<pre><code><span class="co"># Run FigR</span>
<span class="va">fig.d</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/runFigRGRN.html">runFigRGRN</a></span><span class="op">(</span>ATAC.se<span class="op">=</span>,<span class="va">ATAC.SE</span>,
                    rnaMat<span class="op">=</span><span class="va">rnaMat.smooth</span>, <span class="co"># Smoothed RNA matrix using paired cell kNNs</span>
                    dorcMat<span class="op">=</span><span class="va">dorcMat.smooth</span>,
                    dorcTab<span class="op">=</span><span class="va">cisCor.filt</span>,
                    genome<span class="op">=</span><span class="st">"hg19"</span>,
                    dorcGenes<span class="op">=</span><span class="va">dorcGenes</span>,
                    nCores<span class="op">=</span><span class="fl">4</span><span class="op">)</span></code></pre>
</div>
<div class="section level3">
<h3 id="visualizing-figr-results">Visualizing FigR results<a class="anchor" aria-label="anchor" href="#visualizing-figr-results"></a>
</h3>
<pre><code><span class="co"># Visualize all TF-DORC regulation scores (Scatter plot)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caleblareau/BuenColors" class="external-link">BuenColors</a></span><span class="op">)</span> <span class="co"># https://github.com/caleblareau/BuenColors</span>

<span class="va">fig.d</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">Corr.log10P</span>,<span class="va">Enrichment.log10P</span>,color<span class="op">=</span><span class="va">Score</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu">geom_point_rast</span><span class="op">(</span>size<span class="op">=</span><span class="fl">0.01</span>,shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span> 
          <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_color_gradientn</a></span><span class="op">(</span>colours <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BuenColors/man/jdb_palette.html" class="external-link">jdb_palette</a></span><span class="op">(</span><span class="st">"solar_extra"</span><span class="op">)</span>,limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">4</span>,<span class="fl">4</span><span class="op">)</span>,oob <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org/reference/oob.html" class="external-link">squish</a></span><span class="op">)</span></code></pre>
</div>
</div>
<div class="section level1">
<h1 id="about">About<a class="anchor" aria-label="anchor" href="#about"></a>
</h1>
<p>We are actively developing this package and will include more detailed vignettes soon!</p>
<p>Check out our walkthroughs on applying the FigR framework to multi-modal and independently derived scATAC-seq / scRNA-seq data by clicking on the links below:</p>
<p align="left">
</p>
<p><a href="https://htmlpreview.github.io/?https://raw.githubusercontent.com/buenrostrolab/FigR/master/vignettes/web_only/FigR_shareseq_tutorial.html" class="external-link"><img src="reference/figures/skinv2.png" title="FigR on SHARE-seq mouse skin data" height="30%" width="35%"></a></p>

<p align="center">
<a href="https://htmlpreview.github.io/?https://raw.githubusercontent.com/buenrostrolab/FigR/master/vignettes/web_only/FigR_stim_tutorial.html" class="external-link"><img src="reference/figures/PBMCs.png" title="FigR on independently assayed PBMC data" height="30%" width="35%"></a>
</p>
</div>
<div class="section level1">
<h1 id="reference">Reference<a class="anchor" aria-label="anchor" href="#reference"></a>
</h1>
<p>See our <a href="https://www.biorxiv.org/content/10.1101/2021.07.28.453784v1" class="external-link">recent preprint</a> detailing the use of the FigR framework for gene regulatory network inference using scATAC and scRNA-seq integration</p>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/buenrostrolab/FigR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/buenrostrolab/FigR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing FigR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Vinay Kartha <br><small class="roles"> Maintainer </small>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Vinay Kartha.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
