<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="FigR">
<title>Application of FigR to SHARE-seq skin data • FigR</title>
<script src="../deps/jquery-3.5.1/jquery-3.5.1.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Application of FigR to SHARE-seq skin data">
<meta property="og:description" content="FigR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">FigR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-tutorials">Tutorials</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-tutorials">
    <a class="dropdown-item" href="../articles/FigR_stim.html">Pairing scATAC and scRNA-seq</a>
    <a class="dropdown-item" href="../articles/FigR_shareseq.html">FigR on multi-modal data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">
    <span class="fa fa-gears"></span>
     
    Functions
  </a>
</li>
<h6 class="dropdown-header" data-toc-skip>News</h6>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/buenrostrolab/FigR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="FigR_shareseq_files/htmlwidgets-1.5.3/htmlwidgets.js"></script><script src="FigR_shareseq_files/plotly-binding-4.9.3/plotly.js"></script><script src="FigR_shareseq_files/typedarray-0.1/typedarray.min.js"></script><link href="FigR_shareseq_files/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet">
<script src="FigR_shareseq_files/crosstalk-1.1.1/js/crosstalk.min.js"></script><link href="FigR_shareseq_files/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="FigR_shareseq_files/plotly-main-1.57.1/plotly-latest.min.js"></script><script src="FigR_shareseq_files/d3-4.5.0/d3.min.js"></script><script src="FigR_shareseq_files/forceNetwork-binding-0.4/forceNetwork.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Application of FigR to SHARE-seq skin data</h1>
                        <h4 data-toc-skip class="author">Vinay Kartha</h4>
            
            <h4 data-toc-skip class="date">2022-03-26</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/buenrostrolab/FigR/blob/HEAD/vignettes/FigR_shareseq.Rmd" class="external-link"><code>vignettes/FigR_shareseq.Rmd</code></a></small>
      <div class="d-none name"><code>FigR_shareseq.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="running-figr-on-multi-modal-data-e-g--share-seq-10x-multi-ome">Running FigR on multi-modal data (e.g. SHARE-seq / 10x multi-ome)<a class="anchor" aria-label="anchor" href="#running-figr-on-multi-modal-data-e-g--share-seq-10x-multi-ome"></a>
</h2>
<p>Here we demo the application of the FigR workflow <a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;Kartha VK et al. Functional Inference of Gene Regulation using Single-Cell Multi-Omics. 2021 &lt;a href="https://www.biorxiv.org/content/10.1101/2021.07.28.453784v1" class="external-link"&gt;(pre-print)&lt;/a&gt;&lt;/p&gt;'><sup>1</sup></a> to previously published SHARE-seq data <a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;Ma S et al. Chromatin Potential Identified by Shared Single-Cell Profiling of RNA and Chromatin. &lt;a href="https://www.sciencedirect.com/science/article/pii/S0092867420312538" class="external-link"&gt;&lt;em&gt;Cell&lt;/em&gt; 2020&lt;/a&gt;&lt;/p&gt;'><sup>2</sup></a></p>
<div class="section level3">
<h3 id="data-downloads-pre-processing">Data downloads / Pre-processing<a class="anchor" aria-label="anchor" href="#data-downloads-pre-processing"></a>
</h3>
<p>The data used in this tutorial can be downloaded (WARNING: Downloading this data will take up ~ 0.5 GB of your hard drive space) as shown below. We then set the unzipped directory (destination folder) to be the working directory, prior to loading the R objects.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Directory where data will be downloaded to</span>
<span class="va">workingDir</span> <span class="op">&lt;-</span> <span class="st">"/Users/vinay/Documents/FigR_package/shareseq"</span>
<span class="va">shareseqZip</span> <span class="op">&lt;-</span> <span class="st">"https://s3.us-east-1.amazonaws.com/vkartha/FigR/FigR_SHAREseq.zip"</span>
<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="va">shareseqZip</span>,
              destfile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">workingDir</span>,<span class="fu"><a href="https://rdrr.io/r/base/basename.html" class="external-link">basename</a></span><span class="op">(</span><span class="va">shareseqZip</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/unzip.html" class="external-link">unzip</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">workingDir</span>,<span class="fu"><a href="https://rdrr.io/r/base/basename.html" class="external-link">basename</a></span><span class="op">(</span><span class="va">shareseqZip</span><span class="op">)</span>,sep<span class="op">=</span><span class="st">"/"</span><span class="op">)</span>,exdir <span class="op">=</span> <span class="va">workingDir</span>,overwrite <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">setwd</a></span><span class="op">(</span><span class="va">workingDir</span><span class="op">)</span></code></pre></div>
<p>The first step of FigR is to determine peak-gene associations using paired single cell accessibility (scATAC-seq) and RNA (scRNA-seq) count data</p>
<p>First, let us load some dependencies, including the main FigR package</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">FNN</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; Warning: package 'FNN' was built under R version 4.0.4</span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">chromVAR</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">doParallel</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caleblareau/BuenColors" class="external-link">BuenColors</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://buenrostrolab.github.io/FigR/">FigR</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; Warning: package 'cowplot' was built under R version 4.0.4</span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BSgenome.Mmusculus.UCSC.mm10</span><span class="op">)</span></code></pre></div>
<p>We require scATAC-seq peak counts as a <code>SummarizedExperiment</code> object, where the <code>rowRanges</code> are fixed-width accessibility peak ranges (typically 300 bp or so), and the <code>assay</code> object is a <code>sparseMatrix</code> of Tn5-instertion counts per peak per cell. Additionally, the scRNA-seq matrix can be provided as a <code>sparseMatrix</code> of the log-normalized gene expression levels for the same cells (e.g. processed using Seurat)</p>
<p>In this example, we show these two inputs pertaining to a SHARE-seq experiment performed on mouse skin cells. For simplicity, we only run this on a subset of cells (quality filtered, and sampled to n=10,000 cells for faster run-time). As a result, note that the results will be similar to what was presented in Kartha et al. , but not identical.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ATAC.se</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"./shareseq_skin_SE_final.rds"</span><span class="op">)</span>
<span class="va">RNAmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"./shareseq_skin_RNAnorm_final.rds"</span><span class="op">)</span> <span class="co"># Normalized</span>

<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">ATAC.se</span><span class="op">)</span> <span class="co"># Peaks x Cells</span></code></pre></div>
<pre><code><span class="co">#&gt; [1] 344592  34774</span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">RNAmat</span><span class="op">)</span> <span class="co"># Genes x Cells</span></code></pre></div>
<pre><code><span class="co">#&gt; [1] 19303 34774</span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>
<span class="va">cellsToKeep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">ATAC.se</span><span class="op">)</span>,size <span class="op">=</span> <span class="fl">10000</span>,replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="va">ATAC.se</span> <span class="op">&lt;-</span> <span class="va">ATAC.se</span><span class="op">[</span>,<span class="va">cellsToKeep</span><span class="op">]</span>
<span class="va">RNAmat</span> <span class="op">&lt;-</span> <span class="va">RNAmat</span><span class="op">[</span>,<span class="va">cellsToKeep</span><span class="op">]</span>

<span class="co"># Remove genes with zero expression across all cells</span>
<span class="va">RNAmat</span> <span class="op">&lt;-</span> <span class="va">RNAmat</span><span class="op">[</span><span class="fu">Matrix</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">RNAmat</span><span class="op">)</span><span class="op">!=</span><span class="fl">0</span>,<span class="op">]</span></code></pre></div>
<p>Additionally, we load an matrix object of cisTopic probability scores that was generated as part of the dimensionality reduction process implemented on the scATAC-seq count data using the <a href="https://github.com/aertslab/cisTopic" class="external-link">cisTopic</a> R package. This will enable us to derive a cell kNN matrix (k-nearest neighbors graph) that is used for smoothing the data. Note, if you used an alternative method for dimensionality reduction, you can derive this cell kNN matrix using alternative inputs (e.g. using the top N LSI / PCA components)</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cisAssign</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"./shareseq_skin_cisTopicPs.rds"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">cisAssign</span><span class="op">)</span> <span class="co"># Cells x Topics</span>

<span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="va">cellsToKeep</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">cisAssign</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Subset</span>
<span class="va">cisAssign</span> <span class="op">&lt;-</span> <span class="va">cisAssign</span><span class="op">[</span><span class="va">cellsToKeep</span>,<span class="op">]</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Derive cell kNN using this</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>
<span class="va">cellkNN</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/FNN/man/get.knn.html" class="external-link">get.knn</a></span><span class="op">(</span><span class="va">cisAssign</span>,k <span class="op">=</span> <span class="fl">30</span><span class="op">)</span><span class="op">$</span><span class="va">nn.index</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">cellkNN</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">cellkNN</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">cellsToKeep</span></code></pre></div>
<p>We can use previously computed UMAP embeddings and cell type annotations to take a look at what the data encompasses.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">annoCols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"./shareseq_skin_annoCols.rds"</span><span class="op">)</span>

<span class="co"># Plot</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">ATAC.se</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">UMAP1</span>,<span class="va">UMAP2</span>,color<span class="op">=</span><span class="va">cellAnnot</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="va">annoCols</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span>override.aes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="FigR_shareseq_files/figure-html/firstUMAP-1.png" width="768" style="display: block; margin: auto;"></p>
<p>Now, we are ready to run the peak-gene correlations across cells (see below)</p>
</div>
<div class="section level3">
<h3 id="peak-gene-association-testing">Peak-gene association testing<a class="anchor" aria-label="anchor" href="#peak-gene-association-testing"></a>
</h3>
<p>Next, we take a default (10 kb) window around each gene’s TSS, and compute the Spearman correlation across all cells between their peak accessibility counts (mean-centered) and the normalized RNA expression. For each peak-gene pair correlation, we use (default <code>n=100</code>) background peaks matched for GC content and accessibility to correlate to the same gene, so that we can test for significance (permutation p-value).</p>
<blockquote>
<blockquote>
<p>This works a lot faster when supported by parallelization (due to intensive background correlation calculations). Below, we use 4 cores. For 10k cells, 478,254 resulting gene-peak pairs and 100 background calculations per pair, it takes ~ 11.8 hrs (so be warned!). We recommend running this next chunk separately using <code>Rscript</code> (outside of Rstudio) and saving the resulting table, so that it can be loaded for the next steps</p>
</blockquote>
</blockquote>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Don't run interactively</span>
<span class="va">cisCorr</span> <span class="op">&lt;-</span> <span class="fu">FigR</span><span class="fu">::</span><span class="fu"><a href="../reference/runGenePeakcorr.html">runGenePeakcorr</a></span><span class="op">(</span>ATAC.se <span class="op">=</span> <span class="va">ATAC.se</span>,
                           RNAmat <span class="op">=</span> <span class="va">RNAmat</span>,
                           genome <span class="op">=</span> <span class="st">"mm10"</span>, <span class="co"># One of hg19, mm10 or hg38 </span>
                           nCores <span class="op">=</span> <span class="fl">4</span>,
                           p.cut <span class="op">=</span> <span class="cn">NULL</span>, <span class="co"># Set this to NULL and we can filter later</span>
                           n_bg <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cisCorr</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="determining-dorcs">Determining DORCs<a class="anchor" aria-label="anchor" href="#determining-dorcs"></a>
</h3>
<p>We can now filter correlations based on the background p-value, and summarize the genes that have a relatively high number of significant peak-gene associations (i.e. Domains of regulatory chromatin or ‘DORCs’)</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cisCorr.filt</span> <span class="op">&lt;-</span> <span class="va">cisCorr</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pvalZ</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span>

<span class="va">dorcGenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dorcJPlot.html">dorcJPlot</a></span><span class="op">(</span>dorcTab <span class="op">=</span> <span class="va">cisCorr.filt</span>,
                         cutoff <span class="op">=</span> <span class="fl">10</span>, <span class="co"># No. sig peaks needed to be called a DORC</span>
                         labelTop <span class="op">=</span> <span class="fl">20</span>,
                         returnGeneList <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Set this to FALSE for just the plot</span>
                         force<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="FigR_shareseq_files/figure-html/getDORCs-1.png" width="576" style="display: block; margin: auto;"></p>
<p>We can also just get the full ranked table of genes and number of significantly associated peaks, instead of only getting the gene names (so you can threshold yourself)</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Unfiltered</span>
<span class="va">numDorcs</span> <span class="op">&lt;-</span> <span class="va">cisCorr.filt</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Gene</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">tally</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span>
<span class="va">numDorcs</span></code></pre></div>
<pre><code><span class="co">#&gt; <span style="color: #949494;"># A tibble: 12,765 x 2</span></span>
<span class="co">#&gt;    Gene       n</span>
<span class="co">#&gt;    <span style="color: #949494;font-style: italic;">&lt;chr&gt;</span><span>  </span><span style="color: #949494;font-style: italic;">&lt;int&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span><span> Notch1    50</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span><span> Atf3      48</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span><span> Gfpt2     41</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span><span> Col1a2    40</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span><span> Col3a1    37</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span><span> Ifrd1     37</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span><span> Emp1      36</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span><span> Thbs1     36</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span><span> Cyr61     33</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span><span> Ezr       33</span></span>
<span class="co">#&gt; <span style="color: #949494;"># ... with 12,755 more rows</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="visualizing-dorcs">Visualizing DORCs<a class="anchor" aria-label="anchor" href="#visualizing-dorcs"></a>
</h3>
<p>To get the DORC accessibility scores, we simply sum up the chromatin accessibility peak counts for peaks associated with a given gene.</p>
<blockquote>
<blockquote>
<p>NOTE: It is crucial that you use the exact same <code>SummarizedExperiment</code> object here as the one that was used in the <code>runGenePeakcorr</code> step</p>
</blockquote>
</blockquote>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dorcMat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDORCScores.html">getDORCScores</a></span><span class="op">(</span>ATAC.se <span class="op">=</span> <span class="va">ATAC.se</span>, <span class="co"># Has to be same SE as used in previous step</span>
                         dorcTab <span class="op">=</span> <span class="va">cisCorr.filt</span>,
                         geneList <span class="op">=</span> <span class="va">dorcGenes</span>,
                         nCores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; ........</span>
<span class="co">#&gt; Normalizing scATAC counts ..</span>
<span class="co">#&gt; Centering counts for cells sequentially in groups of size  5000  ..</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Computing centered counts for cells:  1  to  5000 ..</span>
<span class="co">#&gt; Computing centered counts per cell using mean reads in features ..</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Computing centered counts for cells:  5001  to  10000 ..</span>
<span class="co">#&gt; Computing centered counts per cell using mean reads in features ..</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Merging results..</span>
<span class="co">#&gt; Done!</span>
<span class="co">#&gt; Computing DORC scores ..</span>
<span class="co">#&gt; Running in parallel using  1 cores ..</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Time Elapsed:  2.06359885136286 mins</span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">dorcMat</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; [1]   586 10000</span></code></pre>
<p>We can then smooth these (sparse) counts, and visualize side-by-side with the expression.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Smooth dorc scores using cell KNNs (k=30)</span>
<span class="va">dorcMat.s</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/smoothScoresNN.html">smoothScoresNN</a></span><span class="op">(</span>NNmat <span class="op">=</span> <span class="va">cellkNN</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">]</span>,mat <span class="op">=</span> <span class="va">dorcMat</span>,nCores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>

<span class="co"># Smooth RNA using cell KNNs</span>
<span class="co"># This takes longer since it's all genes</span>
<span class="va">RNAmat.s</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/smoothScoresNN.html">smoothScoresNN</a></span><span class="op">(</span>NNmat <span class="op">=</span> <span class="va">cellkNN</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">]</span>,mat <span class="op">=</span> <span class="va">RNAmat</span>,nCores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Visualize on pre-computed UMAP</span>
<span class="va">umap.d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">ATAC.se</span><span class="op">)</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UMAP1"</span>,<span class="st">"UMAP2"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>

<span class="co"># DORC score for Dlx3</span>
<span class="va">dorcg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotMarker2D.html">plotMarker2D</a></span><span class="op">(</span><span class="va">umap.d</span>,<span class="va">dorcMat.s</span>,markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Dlx3"</span><span class="op">)</span>,maxCutoff <span class="op">=</span> <span class="st">"q0.99"</span>,colorPalette <span class="op">=</span> <span class="st">"brewer_heat"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Dlx3 DORC"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; Plotting  Dlx3</span></code></pre>
<pre><code><span class="co">#&gt; Warning: package 'ggrastr' was built under R version 4.0.4</span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># RNA for Dlx3</span>
<span class="va">rnag</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotMarker2D.html">plotMarker2D</a></span><span class="op">(</span><span class="va">umap.d</span>,<span class="va">RNAmat.s</span>,markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Dlx3"</span><span class="op">)</span>,maxCutoff <span class="op">=</span> <span class="st">"q0.99"</span>,colorPalette <span class="op">=</span> <span class="st">"brewer_purple"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Dlx3 RNA"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; Plotting  Dlx3</span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; Warning: package 'patchwork' was built under R version 4.0.4</span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dorcg</span> <span class="op">+</span> <span class="va">rnag</span></code></pre></div>
<p><img src="FigR_shareseq_files/figure-html/visDORCs-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="tf-gene-associations">TF-gene associations<a class="anchor" aria-label="anchor" href="#tf-gene-associations"></a>
</h3>
<p>The core component of FigR is to determine TFs that are putative regulators (acivators or repressors) of DORCs. By specifying a built-in reference motif database (see Methods of associated manuscript), and providing smoothed DORC accessibility and RNA count matrices as input, we determine which DORCs are enriched for different TF binding motifs, in addition to testing their correlations to TF RNA expression. This can be done using a single command in <code>FigR</code></p>
<blockquote>
<blockquote>
<p>For 708 TFs against 586 DORCs (testing with n=50 background iterations), this took ~ 1.5 hrs using 4 cores (for the same 10k cells).</p>
</blockquote>
</blockquote>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">figR.d</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runFigRGRN.html">runFigRGRN</a></span><span class="op">(</span>ATAC.se <span class="op">=</span> <span class="va">ATAC.se</span>, <span class="co"># Must be the same input as used in runGenePeakcorr()</span>
                     dorcTab <span class="op">=</span> <span class="va">cisCorr.filt</span>, <span class="co"># Filtered peak-gene associations</span>
                     genome <span class="op">=</span> <span class="st">"mm10"</span>,
                     dorcMat <span class="op">=</span> <span class="va">dorcMat.s</span>,
                     rnaMat <span class="op">=</span> <span class="va">RNAmat.s</span>, 
                     nCores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visualizing-figr-results">Visualizing FigR results<a class="anchor" aria-label="anchor" href="#visualizing-figr-results"></a>
</h3>
<p>There are a few ways we can visualize the resulting TF-DORC associations inferred from FigR, that vary based on thresholding and the type of layout choice</p>
<div class="section level4">
<h4 id="global-regulation-profile">Global regulation profile<a class="anchor" aria-label="anchor" href="#global-regulation-profile"></a>
</h4>
<p>First, we can simply plot all TF-DORC associations (all tested TFs by all tested DORCs) as a scatter plot, where each point is a TF-DORC pair. The x-axis provides the significance of the correlation of each TF’s expression to the DORC accessibility (Z-test relative to background), and the y-axis shows the relative enrichment of the TF’s motif among the DORC-associated peaks. Points are colored by their determined FigR regulation score. Putative activating associations are highlighted in red, while repressive associations are in blue</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">figR.d</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">Corr.log10P</span>,<span class="va">Enrichment.log10P</span>,color<span class="op">=</span><span class="va">Score</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">ggrastr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggrastr/man/geom_point_rast.html" class="external-link">geom_point_rast</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">0.01</span>,shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_color_gradientn</a></span><span class="op">(</span>colours <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BuenColors/man/jdb_palette.html" class="external-link">jdb_palette</a></span><span class="op">(</span><span class="st">"solar_extra"</span><span class="op">)</span>,limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>,<span class="fl">3</span><span class="op">)</span>,oob <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org/reference/oob.html" class="external-link">squish</a></span>,breaks<span class="op">=</span><span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org/reference/breaks_pretty.html" class="external-link">breaks_pretty</a></span><span class="op">(</span>n<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="FigR_shareseq_files/figure-html/plotallFigR-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="ranking-tf-drivers">Ranking TF drivers<a class="anchor" aria-label="anchor" href="#ranking-tf-drivers"></a>
</h4>
<p>We can rank TFs by the overall (mean) regulation score across all DORCs. This helps give a snapshot of the major activators and repressors across all cells in the data given the DORCs</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/rankDrivers.html">rankDrivers</a></span><span class="op">(</span><span class="va">figR.d</span>,rankBy <span class="op">=</span> <span class="st">"meanScore"</span>,interactive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; Warning: ggrepel: 50 unlabeled data points (too many overlaps). Consider</span>
<span class="co">#&gt; increasing max.overlaps</span></code></pre>
<p><img src="FigR_shareseq_files/figure-html/rankDrivers-1.png" width="700"></p>
<blockquote>
<p>Specify the <code>myLabels</code> parameter to label the plots above for select TFs (and not just those in the top and bottom percentiles by default). Useful to see where your TFs of interest fall among all TFs</p>
</blockquote>
<p>We can also rank by the total number of targets per TF, passing a specified score filter</p>
<blockquote>
<p>You can set the <code>interactive</code> parameter to <code>TRUE</code> and then hover over the bars for more detail, in cases where it’s hard to tell from the labels or axes (if plotting too many TFs)</p>
</blockquote>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/rankDrivers.html">rankDrivers</a></span><span class="op">(</span><span class="va">figR.d</span>,score.cut <span class="op">=</span> <span class="fl">2</span>,rankBy <span class="op">=</span> <span class="st">"nTargets"</span>,interactive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-ef51b2bc1748017b5bab" style="width:700px;height:384px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-ef51b2bc1748017b5bab">{"x":{"data":[{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9,0.9,0.9,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.900000000000002,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006],"base":[0,0,0,0,0,0,1,3,0,0,0,0,0,0,0,0,0,0,0,8,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,-2,0,0,0,0,0,0,0,-5,0,0,0,0,-8,-5,0,0,0,0,0,0,-16,0,0],"x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64],"y":[77,60,55,48,27,27,24,23,19,9,6,6,5,5,5,4,4,4,3,11,2,3,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,0,0,0,0,2,0,0,0,0,3,1,0,0,0,0,0,0,1,0,0],"text":["Motif: Lef1<br />numActivated: 77<br />numRepressed:  0","Motif: Gata6<br />numActivated: 60<br />numRepressed:  0","Motif: Egr2<br />numActivated: 55<br />numRepressed:  0","Motif: Dlx2<br />numActivated: 48<br />numRepressed:  0","Motif: Dlx3<br />numActivated: 27<br />numRepressed:  0","Motif: Sox9<br />numActivated: 27<br />numRepressed:  0","Motif: Prdm5<br />numActivated: 24<br />numRepressed:  1","Motif: Tcf4<br />numActivated: 23<br />numRepressed:  3","Motif: Klf4<br />numActivated: 19<br />numRepressed:  0","Motif: Nfatc1<br />numActivated:  9<br />numRepressed:  0","Motif: Egr1<br />numActivated:  6<br />numRepressed:  0","Motif: Mef2c<br />numActivated:  6<br />numRepressed:  0","Motif: Ahr<br />numActivated:  5<br />numRepressed:  0","Motif: Fosb<br />numActivated:  5<br />numRepressed:  0","Motif: Grhl1<br />numActivated:  5<br />numRepressed:  0","Motif: Fos<br />numActivated:  4<br />numRepressed:  0","Motif: Gli2<br />numActivated:  4<br />numRepressed:  0","Motif: Msx2<br />numActivated:  4<br />numRepressed:  0","Motif: Fli1<br />numActivated:  3<br />numRepressed:  0","Motif: Nfkb1<br />numActivated: 11<br />numRepressed:  8","Motif: Klf9<br />numActivated:  2<br />numRepressed:  0","Motif: Nfia<br />numActivated:  3<br />numRepressed:  1","Motif: Arid2<br />numActivated:  1<br />numRepressed:  0","Motif: Creb3l1<br />numActivated:  1<br />numRepressed:  0","Motif: Ebf3<br />numActivated:  1<br />numRepressed:  0","Motif: Elf2<br />numActivated:  1<br />numRepressed:  0","Motif: Foxc1<br />numActivated:  1<br />numRepressed:  0","Motif: Irf1<br />numActivated:  1<br />numRepressed:  0","Motif: Klf12<br />numActivated:  1<br />numRepressed:  0","Motif: Nfat5<br />numActivated:  1<br />numRepressed:  0","Motif: Nfic<br />numActivated:  1<br />numRepressed:  0","Motif: Nr4a1<br />numActivated:  1<br />numRepressed:  0","Motif: Sox18<br />numActivated:  1<br />numRepressed:  0","Motif: Sox5<br />numActivated:  1<br />numRepressed:  0","Motif: Sox7<br />numActivated:  1<br />numRepressed:  0","Motif: Tcf12<br />numActivated:  1<br />numRepressed:  0","Motif: Tcf7<br />numActivated:  1<br />numRepressed:  0","Motif: Vdr<br />numActivated:  1<br />numRepressed:  0","Motif: Ybx3<br />numActivated:  1<br />numRepressed:  0","Motif: Zbtb7c<br />numActivated:  1<br />numRepressed:  0","Motif: Setbp1<br />numActivated:  1<br />numRepressed:  1","Motif: Clock<br />numActivated:  0<br />numRepressed:  1","Motif: Ehf<br />numActivated:  0<br />numRepressed:  1","Motif: Ikzf2<br />numActivated:  0<br />numRepressed:  1","Motif: Mef2a<br />numActivated:  0<br />numRepressed:  1","Motif: Nfatc2<br />numActivated:  0<br />numRepressed:  1","Motif: Rel<br />numActivated:  0<br />numRepressed:  1","Motif: Rfx2<br />numActivated:  0<br />numRepressed:  1","Motif: Sox6<br />numActivated:  2<br />numRepressed:  3","Motif: Zeb2<br />numActivated:  0<br />numRepressed:  1","Motif: En1<br />numActivated:  0<br />numRepressed:  2","Motif: Gli3<br />numActivated:  0<br />numRepressed:  2","Motif: Klf3<br />numActivated:  0<br />numRepressed:  2","Motif: Klf6<br />numActivated:  3<br />numRepressed:  5","Motif: Bcl11a<br />numActivated:  1<br />numRepressed:  4","Motif: Rora<br />numActivated:  0<br />numRepressed:  3","Motif: Tcf7l1<br />numActivated:  0<br />numRepressed:  3","Motif: Tfap2b<br />numActivated:  0<br />numRepressed:  3","Motif: Trps1<br />numActivated:  0<br />numRepressed:  3","Motif: Tcf7l2<br />numActivated:  0<br />numRepressed:  4","Motif: Klf5<br />numActivated:  0<br />numRepressed: 10","Motif: Ets2<br />numActivated:  1<br />numRepressed: 15","Motif: Bcl11b<br />numActivated:  0<br />numRepressed: 15","Motif: Barx2<br />numActivated:  0<br />numRepressed: 29"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(205,38,38,1)","line":{"width":0.377952755905512,"color":"rgba(211,211,211,1)"}},"name":"numActivatedY","legendgroup":"numActivatedY","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9,0.9,0.9,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.900000000000002,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006],"base":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,-1,-1,-1,-1,-1,-1,-1,-1,-3,-1,-2,-2,-2,-5,-4,-3,-3,-3,-3,-4,-10,-15,-15,-29],"x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64],"y":[0,0,0,0,0,0,1,3,0,0,0,0,0,0,0,0,0,0,0,8,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,1,1,1,1,1,3,1,2,2,2,5,4,3,3,3,3,4,10,15,15,29],"text":["Motif: Lef1<br />numActivated: 77<br />numRepressed:  0","Motif: Gata6<br />numActivated: 60<br />numRepressed:  0","Motif: Egr2<br />numActivated: 55<br />numRepressed:  0","Motif: Dlx2<br />numActivated: 48<br />numRepressed:  0","Motif: Dlx3<br />numActivated: 27<br />numRepressed:  0","Motif: Sox9<br />numActivated: 27<br />numRepressed:  0","Motif: Prdm5<br />numActivated: 24<br />numRepressed:  1","Motif: Tcf4<br />numActivated: 23<br />numRepressed:  3","Motif: Klf4<br />numActivated: 19<br />numRepressed:  0","Motif: Nfatc1<br />numActivated:  9<br />numRepressed:  0","Motif: Egr1<br />numActivated:  6<br />numRepressed:  0","Motif: Mef2c<br />numActivated:  6<br />numRepressed:  0","Motif: Ahr<br />numActivated:  5<br />numRepressed:  0","Motif: Fosb<br />numActivated:  5<br />numRepressed:  0","Motif: Grhl1<br />numActivated:  5<br />numRepressed:  0","Motif: Fos<br />numActivated:  4<br />numRepressed:  0","Motif: Gli2<br />numActivated:  4<br />numRepressed:  0","Motif: Msx2<br />numActivated:  4<br />numRepressed:  0","Motif: Fli1<br />numActivated:  3<br />numRepressed:  0","Motif: Nfkb1<br />numActivated: 11<br />numRepressed:  8","Motif: Klf9<br />numActivated:  2<br />numRepressed:  0","Motif: Nfia<br />numActivated:  3<br />numRepressed:  1","Motif: Arid2<br />numActivated:  1<br />numRepressed:  0","Motif: Creb3l1<br />numActivated:  1<br />numRepressed:  0","Motif: Ebf3<br />numActivated:  1<br />numRepressed:  0","Motif: Elf2<br />numActivated:  1<br />numRepressed:  0","Motif: Foxc1<br />numActivated:  1<br />numRepressed:  0","Motif: Irf1<br />numActivated:  1<br />numRepressed:  0","Motif: Klf12<br />numActivated:  1<br />numRepressed:  0","Motif: Nfat5<br />numActivated:  1<br />numRepressed:  0","Motif: Nfic<br />numActivated:  1<br />numRepressed:  0","Motif: Nr4a1<br />numActivated:  1<br />numRepressed:  0","Motif: Sox18<br />numActivated:  1<br />numRepressed:  0","Motif: Sox5<br />numActivated:  1<br />numRepressed:  0","Motif: Sox7<br />numActivated:  1<br />numRepressed:  0","Motif: Tcf12<br />numActivated:  1<br />numRepressed:  0","Motif: Tcf7<br />numActivated:  1<br />numRepressed:  0","Motif: Vdr<br />numActivated:  1<br />numRepressed:  0","Motif: Ybx3<br />numActivated:  1<br />numRepressed:  0","Motif: Zbtb7c<br />numActivated:  1<br />numRepressed:  0","Motif: Setbp1<br />numActivated:  1<br />numRepressed:  1","Motif: Clock<br />numActivated:  0<br />numRepressed:  1","Motif: Ehf<br />numActivated:  0<br />numRepressed:  1","Motif: Ikzf2<br />numActivated:  0<br />numRepressed:  1","Motif: Mef2a<br />numActivated:  0<br />numRepressed:  1","Motif: Nfatc2<br />numActivated:  0<br />numRepressed:  1","Motif: Rel<br />numActivated:  0<br />numRepressed:  1","Motif: Rfx2<br />numActivated:  0<br />numRepressed:  1","Motif: Sox6<br />numActivated:  2<br />numRepressed:  3","Motif: Zeb2<br />numActivated:  0<br />numRepressed:  1","Motif: En1<br />numActivated:  0<br />numRepressed:  2","Motif: Gli3<br />numActivated:  0<br />numRepressed:  2","Motif: Klf3<br />numActivated:  0<br />numRepressed:  2","Motif: Klf6<br />numActivated:  3<br />numRepressed:  5","Motif: Bcl11a<br />numActivated:  1<br />numRepressed:  4","Motif: Rora<br />numActivated:  0<br />numRepressed:  3","Motif: Tcf7l1<br />numActivated:  0<br />numRepressed:  3","Motif: Tfap2b<br />numActivated:  0<br />numRepressed:  3","Motif: Trps1<br />numActivated:  0<br />numRepressed:  3","Motif: Tcf7l2<br />numActivated:  0<br />numRepressed:  4","Motif: Klf5<br />numActivated:  0<br />numRepressed: 10","Motif: Ets2<br />numActivated:  1<br />numRepressed: 15","Motif: Bcl11b<br />numActivated:  0<br />numRepressed: 15","Motif: Barx2<br />numActivated:  0<br />numRepressed: 29"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(54,100,139,1)","line":{"width":0.377952755905512,"color":"rgba(211,211,211,1)"}},"name":"numRepressedY","legendgroup":"numRepressedY","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.4,64.6],"y":[0,0],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.3181126331811,"r":7.30593607305936,"b":31.5829528158295,"l":37.2602739726027},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.4,64.6],"tickmode":"array","ticktext":["Lef1","Gata6","Egr2","Dlx2","Dlx3","Sox9","Prdm5","Tcf4","Klf4","Nfatc1","Egr1","Mef2c","Ahr","Fosb","Grhl1","Fos","Gli2","Msx2","Fli1","Nfkb1","Klf9","Nfia","Arid2","Creb3l1","Ebf3","Elf2","Foxc1","Irf1","Klf12","Nfat5","Nfic","Nr4a1","Sox18","Sox5","Sox7","Tcf12","Tcf7","Vdr","Ybx3","Zbtb7c","Setbp1","Clock","Ehf","Ikzf2","Mef2a","Nfatc2","Rel","Rfx2","Sox6","Zeb2","En1","Gli3","Klf3","Klf6","Bcl11a","Rora","Tcf7l1","Tfap2b","Trps1","Tcf7l2","Klf5","Ets2","Bcl11b","Barx2"],"tickvals":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64],"categoryorder":"array","categoryarray":["Lef1","Gata6","Egr2","Dlx2","Dlx3","Sox9","Prdm5","Tcf4","Klf4","Nfatc1","Egr1","Mef2c","Ahr","Fosb","Grhl1","Fos","Gli2","Msx2","Fli1","Nfkb1","Klf9","Nfia","Arid2","Creb3l1","Ebf3","Elf2","Foxc1","Irf1","Klf12","Nfat5","Nfic","Nr4a1","Sox18","Sox5","Sox7","Tcf12","Tcf7","Vdr","Ybx3","Zbtb7c","Setbp1","Clock","Ehf","Ikzf2","Mef2a","Nfatc2","Rel","Rfx2","Sox6","Zeb2","En1","Gli3","Klf3","Klf6","Bcl11a","Rora","Tcf7l1","Tfap2b","Trps1","Tcf7l2","Klf5","Ets2","Bcl11b","Barx2"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":{"text":"Ranked TF Motifs","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-34.3,82.3],"tickmode":"array","ticktext":[30,0,30,60],"tickvals":[-30,0,30,60],"categoryorder":"array","categoryarray":[30,0,30,60],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":{"text":"# Associated genes <br />abs(Score) >= 2","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"4d2c503a73c1":{"x":{},"y":{},"fill":{},"numActivated":{},"numRepressed":{},"type":"bar"},"4d2c6493080":{"yintercept":{}}},"cur_data":"4d2c503a73c1","visdat":{"4d2c503a73c1":["function (y) ","x"],"4d2c6493080":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level4">
<h4 id="scatter-view">Scatter view<a class="anchor" aria-label="anchor" href="#scatter-view"></a>
</h4>
<p>We can also subset the plot above by restricting it to specific TFs. Only points (TFs) passing the user-defined threshold are labeled as putative drivers of the queried DORC</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotDrivers.html">plotDrivers</a></span><span class="op">(</span><span class="va">figR.d</span>,score.cut <span class="op">=</span> <span class="fl">2</span>,marker <span class="op">=</span> <span class="st">"Lef1"</span><span class="op">)</span></code></pre></div>
<p><img src="FigR_shareseq_files/figure-html/plotDrivers-1.png" width="432" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="heatmap-view">Heatmap view<a class="anchor" aria-label="anchor" href="#heatmap-view"></a>
</h4>
<p>For the heatmap plot, we take any TF-DORC association that meets the score cut-off, and plot the union of all TFs (columns) x DORCs (rows), colored by their associated FigR regulation score</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span>
<span class="fu"><a href="../reference/plotfigRHeatmap.html">plotfigRHeatmap</a></span><span class="op">(</span>figR.d <span class="op">=</span> <span class="va">figR.d</span>,
                score.cut <span class="op">=</span> <span class="fl">2</span>,
                column_names_gp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/gpar.html" class="external-link">gpar</a></span><span class="op">(</span>fontsize<span class="op">=</span><span class="fl">6</span><span class="op">)</span>, <span class="co"># from ComplexHeatmap</span>
                show_row_dend <span class="op">=</span> <span class="cn">FALSE</span> <span class="co"># from ComplexHeatmap</span>
                <span class="op">)</span></code></pre></div>
<p><img src="FigR_shareseq_files/figure-html/plotHeat-1.png" width="672" style="display: block; margin: auto;"></p>
<p>You can also restrict the heatmap above to correspond to only specific DORCs or TFs by changing the <code>TF</code> or <code>DORC</code> parameter from <code>NULL</code> (default)</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span>
<span class="fu"><a href="../reference/plotfigRHeatmap.html">plotfigRHeatmap</a></span><span class="op">(</span>figR.d <span class="op">=</span> <span class="va">figR.d</span>,
                score.cut <span class="op">=</span> <span class="fl">1</span>,
                TFs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Lef1"</span>,<span class="st">"Dlx3"</span>,<span class="st">"Grhl1"</span>,<span class="st">"Gata6"</span>,<span class="st">"Klf3"</span>,<span class="st">"Barx2"</span>,<span class="st">"Pou2f3"</span><span class="op">)</span>,
                column_names_gp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/gpar.html" class="external-link">gpar</a></span><span class="op">(</span>fontsize<span class="op">=</span><span class="fl">6</span><span class="op">)</span>, <span class="co"># from ComplexHeatmap</span>
                show_row_dend <span class="op">=</span> <span class="cn">FALSE</span> <span class="co"># from ComplexHeatmap</span>
                <span class="op">)</span></code></pre></div>
<p><img src="FigR_shareseq_files/figure-html/plotHeat2-1.png" width="576" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="network-view">Network view<a class="anchor" aria-label="anchor" href="#network-view"></a>
</h4>
<p>Similarly, we can render the significant associations as a graph network, where each node is a TF/DORC and each edge is an association passing a user-specified threshold</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://CRAN.R-project.org/package=networkD3" class="external-link">networkD3</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; Warning: package 'networkD3' was built under R version 4.0.5</span></code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotfigRNetwork.html">plotfigRNetwork</a></span><span class="op">(</span><span class="va">figR.d</span>,
                score.cut <span class="op">=</span> <span class="fl">2</span>,
                TFs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Lef1"</span>,<span class="st">"Dlx3"</span>,<span class="st">"Grhl1"</span>,<span class="st">"Gata6"</span>,<span class="st">"Klf3"</span>,<span class="st">"Barx2"</span>,<span class="st">"Pou2f3"</span><span class="op">)</span>,
                weight.edges <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-8296205bd7632be1dce0" style="width:700px;height:432.632880098888px;" class="forceNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-8296205bd7632be1dce0">{"x":{"links":{"source":[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,17,18,19,20,21,22,23,24,25,26,27,28,29,30,30,31,32,33,34,34,34,35,36,37,37,38,39,40,41,41,42,43,44,45,46,47,48,49,50,51,52,52,52,53,54,55,56,57,57,58,59,60,60,61,61,62,63,64,64,65,66,67,68,69,69,70,70,71,72,72,73,74,75,76,77,78,79,80,81,81,82,82,83,84,85,85,86,86,86,87,88,89,90,91,92,93,94,95,95,96,96,97,98,99,100,101,102,103,103,104,105,105,106,107,108,108,109,110,111,112,113,114,114,115,116,116,117,117,117,118,119,119,120,121,122,123,124,125,126,127,128,128,129,130,131,132,132,133,134,135,136,137,138,138,138,139,140,140,140,141,141,142,143,143,144,145,146,147,148,149,150,151,152,153,154,155,155,156,156,157,157,158],"target":[159,160,160,160,160,160,160,161,161,160,160,161,160,160,160,162,161,159,161,163,161,161,161,161,160,160,160,160,161,160,160,159,161,160,160,161,159,163,161,161,160,159,163,161,161,160,159,161,160,163,160,160,163,160,160,161,161,160,159,163,161,160,163,160,163,159,163,161,161,163,161,159,161,160,161,159,161,160,161,160,164,159,161,159,161,161,159,161,160,160,161,161,160,160,160,162,159,161,163,161,160,160,159,163,159,163,161,159,160,160,160,164,161,160,161,163,161,163,161,159,161,160,161,161,162,159,161,161,163,161,161,161,159,161,161,160,160,161,161,159,161,161,163,161,159,163,161,161,159,161,160,161,160,160,160,163,161,160,163,161,161,160,163,159,161,161,161,160,160,160,159,163,161,161,159,163,161,163,161,160,163,161,161,162,160,159,160,161,163,161,161,160,160,159,161,163,161,159,161,162],"value":[2.69720241629588,13.5163341127603,13.1886282589664,13.8438868796937,16.8647014809365,14.0952598947386,14.4557104661641,14.234631249286,13.3056061152344,17.311204318878,15.869797295734,14.14283527157,13.4480109404203,15.6366201784333,13.1584839141847,15.2062932695728,13.4568659639978,3.47543132049555,15.1856882428693,13.6222359540841,14.621998555523,13.8453212303766,13.5248063090342,13.2617966296562,14.5318363711396,14.4667595891423,18.7593396739469,17.0868636733464,15.1809086341527,13.8789267448414,14.7554031650496,1.38985759834066,13.6259718422577,14.698742618561,19.9969849994964,13.4269437428221,2.9919318633434,14.0496129962418,13.9965663423276,14.6713716048866,16.1647893217055,3.93587144279897,13.7393767152463,13.3610851282444,14.0232786876551,13.9053842671145,3.12098181732505,15.2807372305405,17.0145042758939,13.7878713019433,14.375410986842,17.6165011221193,13.4349406054849,14.4673817123356,14.1648557102102,13.1673676148798,13.4365439763787,13.7054782235416,3.95070543677867,14.0220963381059,13.3256256196899,16.6086044754666,14.3696136644601,18.7032684363103,13.3409492538512,2.70540915286312,13.9180868739525,14.0847472278804,13.5944727655218,13.1688381406796,13.620807151871,2.94900069614829,14.3888797915291,13.331908506852,14.4074920474862,3.22964322765887,13.3597196691747,13.1807988177212,13.3491581909203,16.4264225514507,3.88654608229054,1.30570878539568,13.6483759696115,3.71696287982978,16.1163846192047,13.4092345955611,3.71055872598652,13.5097106720455,15.2399313761547,18.895679170811,14.3129213839906,15.5833156259678,15.1088739931925,14.4674365465306,18.0641266417937,13.9080510808117,2.16660482349501,14.4528144290301,13.5958350163963,13.399935199978,14.2857846335315,14.7671536893739,3.07357286480706,13.5275815883525,3.94084696433382,13.3453618715315,16.2498168017871,2.96620755966243,14.373748785092,15.1047985366651,13.7269167313748,2.03386736830264,13.4584147501951,13.8715962468407,14.913903082552,14.1475777581632,14.3191114533653,13.4211520387342,15.2170857587923,0,14.7265969693585,15.1802639346943,13.219500392241,14.4238161638235,14.831160234924,3.30992042076646,14.6067573681725,15.2674578032227,13.2375330887949,13.9294885430274,13.9026057377873,15.3978129416813,3.79048323503717,13.1610615881859,13.59782298649,13.5351691314663,15.1183386636158,14.5952601147537,13.2634578416931,3.13695152826195,15.1578070031219,14.2318738764158,13.7682630436811,14.8963506867261,3.09076349908054,13.5193366615164,14.2576847171039,13.3787576651125,3.46056685833272,14.009832586249,15.3860240575015,13.2894109656796,13.3213338214039,14.8492526151394,14.8174652320187,13.39320760615,13.8377080138346,15.6911483379842,14.1383080428363,13.2142500734593,13.4552802928206,14.8845276567863,13.5043133508327,2.70429674529392,13.7678775642054,13.9708503312592,15.7190746390336,14.4223042927143,14.4038002018868,14.8035080564023,2.98122635709914,13.9783145031157,14.6944926093126,13.9150272788245,3.03734251002211,14.2426421197383,13.4637813765676,14.3734055121962,14.8053079102948,15.8875829729971,13.1834581217383,15.9546816232741,15.8951891178783,14.3735243879281,13.9111447312627,2.325438202976,13.5550089921361,13.7606037659293,13.324916819852,14.2703612702753,13.9988879991982,20,13.4459395530555,0.910359907804344,14.228670640654,13.9065346082418,13.8471649700928,2.87434828626689,15.169044098143,18.3131232810377],"colour":["#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#A020F0","#A020F0","#228B22","#A020F0","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#A020F0","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22"]},"nodes":{"name":["1700025G04Rik","4833423E24Rik","9930038B18Rik","A730056A06Rik","Abi3bp","Ablim1","Acsbg1","Acvr1","Adgra3","Adgrl2","Alox12e","Apcdd1","Apoe","Areg","Bace2","Baiap2","Basp1","Bcl2l11","Ccnd2","Cdk17","Chst11","Cnn3","Cntfr","Cpe","Csnk2a2","Cst6","Cstb","Ctdspl","Ctsb","Ctsd","Cux1","Cxadr","Cyp1b1","Dach1","Dcn","Dct","Defb6","Dlx1","Dlx2","Dlx3","Dnase1l3","Dusp2","Ece1","Edar","Eea1","Egr2","Egr3","Ehf","Elovl7","Emid1","Epha2","Fam129b","Fat1","Fbxo32","Fgf5","Gata6","Ggt1","Gja1","Gjb2","Gli2","Gm14321","Gpnmb","Grhl1","Gtf2ird1","H3f3b","Hal","Hist1h1d","Hmgcs2","Hspg2","Hsph1","Hunk","Id1","Id3","Inhbb","Insig2","Jag1","Kif26a","Klk7","Krt16","Krt17","Krt23","Krt25","Krt28","Krt77","Krt79","Lamb1","Lef1","Lgalsl","Lrig1","Lrrc1","Ltbp1","Ltbp4","Map7","Me1","Mgst1","Msx1","Msx2","Mt2","Myo10","Nfatc2","Nfkbie","Notch1","Notch2","Nrp2","Odc1","Padi3","Padi4","Pdcd4","Pfdn2","Pfdn4","Pkib","Plet1","Pmepa1","Ppp2r1b","Prr5l","Ptch1","Ptch2","Ptma","Ptprf","Pxylp1","Qsox1","Ror2","Scel","Sdc4","Sema6a","Sept11","Shb","Skint8","Slc16a6","Slc30a1","Slc37a1","Slc39a10","Slc44a1","Slc7a8","Smad7","Smcp","Sox7","Sox9","Srsf3","St14","Steap2","Tchh","Tec","Tgm6","Tiam2","Tmem136","Tmem45a","Tnfrsf19","Tnip1","Trim2","Tspan13","Ttc7b","Txnrd1","Vangl1","Vdr","Wnt5a","Zfhx3","Zfp36l1","Zfp750","Barx2.","Gata6.","Lef1.","Grhl1.","Dlx3.","Klf3."],"group":["DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","TF","TF","TF","TF","TF","TF"],"nodesize":[8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,3,3,3,3,3,3]},"options":{"NodeID":"name","Group":"group","colourScale":"d3.scaleOrdinal([\" #FF6347\", \"#87CEEB \"])","fontSize":13,"fontFamily":"serif","clickTextSize":32.5,"linkDistance":50,"linkWidth":"function(d) { return Math.sqrt(d.value); }","charge":-15,"opacity":1,"zoom":true,"legend":true,"arrows":false,"nodesize":true,"radiusCalculation":"Math.sqrt(d.nodesize)*2","bounded":true,"opacityNoHover":0.6,"clickAction":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Vinay Kartha, Jason D. Buenrostro.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
