<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="FigR">
<title>Analyzing independently assayed scATAC-seq and scRNA-seq data using FigR • FigR</title>
<script src="../deps/jquery-3.5.1/jquery-3.5.1.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Analyzing independently assayed scATAC-seq and scRNA-seq data using FigR">
<meta property="og:description" content="FigR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">FigR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-tutorials">Tutorials</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-tutorials">
    <a class="dropdown-item" href="../articles/FigR_stim.html">Pairing scATAC and scRNA-seq</a>
    <a class="dropdown-item" href="../articles/FigR_shareseq.html">FigR on multi-modal data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">
    <span class="fa fa-gears"></span>
     
    Functions
  </a>
</li>
<h6 class="dropdown-header" data-toc-skip>News</h6>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/buenrostrolab/FigR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="FigR_stim_files/htmlwidgets-1.5.3/htmlwidgets.js"></script><script src="FigR_stim_files/plotly-binding-4.9.3/plotly.js"></script><script src="FigR_stim_files/typedarray-0.1/typedarray.min.js"></script><link href="FigR_stim_files/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet">
<script src="FigR_stim_files/crosstalk-1.1.1/js/crosstalk.min.js"></script><link href="FigR_stim_files/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="FigR_stim_files/plotly-main-1.57.1/plotly-latest.min.js"></script><script src="FigR_stim_files/d3-4.5.0/d3.min.js"></script><script src="FigR_stim_files/forceNetwork-binding-0.4/forceNetwork.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Analyzing independently assayed scATAC-seq and scRNA-seq data using FigR</h1>
                        <h4 data-toc-skip class="author">Vinay Kartha</h4>
            
            <h4 data-toc-skip class="date">2022-03-13</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/buenrostrolab/FigR/blob/HEAD/vignettes/FigR_stim.Rmd" class="external-link"><code>vignettes/FigR_stim.Rmd</code></a></small>
      <div class="d-none name"><code>FigR_stim.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="running-figr-on-independently-assayed-scatac-seq-and-scrna-seq-data">Running FigR on independently assayed scATAC-seq and scRNA-seq data<a class="anchor" aria-label="anchor" href="#running-figr-on-independently-assayed-scatac-seq-and-scrna-seq-data"></a>
</h2>
<p>In many instances, we don’t have true multi-modal data (i.e. chromatin accessibility and gene expression profiled from the same single cell). In these cases, computationally determining “paired” cells across assays from the same cellular contexts is useful, and is a necessary first step prior to performing downstream analysis, including the <em>cis</em>-regulatory correlation analyses (i.e. peak-to-gene associations) and gene regulatory network inference steps that are part of the FigR framework.</p>
<p>Here we demo the application of the FigR workflow to independently generated scATAC-seq and scRNA-seq data using a subset of the data generated and analyzed in the associated manuscript <a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;Kartha VK et al. Functional Inference of Gene Regulation using Single-Cell Multi-Omics. 2021 &lt;a href="https://www.biorxiv.org/content/10.1101/2021.07.28.453784v1" class="external-link"&gt;(pre-print)&lt;/a&gt;&lt;/p&gt;'><sup>1</sup></a></p>
<div class="section level3">
<h3 id="data-downloads-pre-processing">Data downloads / Pre-processing<a class="anchor" aria-label="anchor" href="#data-downloads-pre-processing"></a>
</h3>
<p>Here, we will use a subset of the PBMC stimulus dataset pertaining to DMSO (Control_1h) condition cells for which we have independently profiled scATAC-seq and scRNA-seq data. The data used in this tutorial can be downloaded (WARNING: Downloading this data will take up ~ 71 MB of your hard drive space) as shown below. We then set the unzipped directory (destination folder) to be the working directory, prior to loading the R objects.</p>
<p>Please see <a href="https://htmlpreview.github.io/?https://raw.githubusercontent.com/buenrostrolab/FigR/master/vignettes/FigR_shareseq_tutorial.html?token=ACJLS4UPFDZVXWXHJIYGXGLBTRSPO" class="external-link">this vignette</a> on applying FigR to multi-modal data for more details on the input object data structures, function parameters, and run time</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Directory where data will be downloaded to</span>
<span class="va">workingDir</span> <span class="op">&lt;-</span> <span class="st">"/Users/vinay/Documents/FigR_package/biorad"</span>
<span class="va">stimZip</span> <span class="op">&lt;-</span> <span class="st">"https://s3.us-east-1.amazonaws.com/vkartha/FigR/FigR_stim.zip"</span>
<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="va">stimZip</span>,
              destfile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">workingDir</span>,<span class="fu"><a href="https://rdrr.io/r/base/basename.html" class="external-link">basename</a></span><span class="op">(</span><span class="va">stimZip</span><span class="op">)</span>,sep <span class="op">=</span> <span class="st">"/"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/unzip.html" class="external-link">unzip</a></span><span class="op">(</span>zipfile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">workingDir</span>,<span class="fu"><a href="https://rdrr.io/r/base/basename.html" class="external-link">basename</a></span><span class="op">(</span><span class="va">stimZip</span><span class="op">)</span>,sep <span class="op">=</span> <span class="st">"/"</span><span class="op">)</span>,exdir <span class="op">=</span> <span class="va">workingDir</span>,overwrite <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">setwd</a></span><span class="op">(</span><span class="va">workingDir</span><span class="op">)</span></code></pre></div>
<p>As before, let us load some dependencies, including the main FigR package</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">doParallel</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caleblareau/BuenColors" class="external-link">BuenColors</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://buenrostrolab.github.io/FigR/">FigR</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; Warning: package 'cowplot' was built under R version 4.0.4</span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BSgenome.Hsapiens.UCSC.hg19</span><span class="op">)</span></code></pre></div>
<p>Here, we require/load the following input objects:</p>
<ul>
<li>A <code>SummarizedExperiment</code> object of peaks x cells, housing the scATAC-seq reads in accessibility peaks raw counts, along with the <code>GRanges</code> of the peaks used as <code>RowRanges</code> info</li>
<li>A normalized <code>sparseMatrix</code> object of genes x cells, housing the scRNA-seq gene counts (e.g. exported from <code>Seurat</code>)</li>
<li>Previously computed canonical correlation analyis (CCA) components (determined using the CCA pipeline in <code>Seurat</code>) for both ATAC and RNA PBMCs</li>
<li>Latent Semantic Index (LSI) components for the scATAC-seq cells, used to derive cell k-nearest neighbors to smooth scores with (NOTE: In the multi-modal vignette, we show how you can use alternative cell embeddings to derive this kNN, such as <code>cisTopic</code> scores. Essentially, you would use whichever reduced dimensions were used to project cells in 2D for visualization)</li>
</ul>
<p>In this example, we show these inputs pertaining to independently generated scATAC and scRNA-seq data in human PBMCs. For simplicity, we only run this on a subset of cells (quality filtered, Control 1h DMSO condition cells for faster run-time). As a result, note that the results will be similar to what was presented in Kartha et al. , but not identical.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ATAC.se</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"./control1h_PBMC_atac_SE.rds"</span><span class="op">)</span>
<span class="va">RNAmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"./control1h_PBMC_RNAnorm.rds"</span><span class="op">)</span>


<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">ATAC.se</span><span class="op">)</span> <span class="co"># Peaks x ATAC cells</span></code></pre></div>
<pre><code><span class="co">#&gt; [1] 219136   5352</span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">RNAmat</span><span class="op">)</span> <span class="co"># Genes x RNA cells</span></code></pre></div>
<pre><code><span class="co">#&gt; [1] 15584  3508</span></code></pre>
<p>Since we’re going to integrate the two datasets here (unpaired as of now), we will use a shared co-embedding that can be generated for both ATAC and RNA cells. See this tutorial for examples of how this can be done. In our case, we used the union of the top variable ATAC genescores (based on a fixed window around each gene’s TSS), and top variable RNA genes across all cells, and performed a CCA using the resulting features and all ATAC/RNA cells as input to the <a href="https://satijalab.org/seurat/reference/runcca" class="external-link">RunCCA</a> function. There are now alternative approaches that are faster / scale better for co-embedding larger datasets, including the <a href="https://satijalab.org/seurat/articles/integration_rpca.html" class="external-link">RPCA framework</a> implemented in the <code>Seurat</code> package.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load CCA components (used for pairing)</span>
<span class="co"># This was derived by running CCA (using all cells) between the original ATAC/RNA data using the top variable scATAC-seq gene scores and scRNA-seq gene expression</span>
<span class="va">CCA_PCs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">'./control1h_PBMC_atac_rna_CCA_l2.rds'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">CCA_PCs</span><span class="op">)</span> <span class="co"># ATAC + RNA (rows), n components (columns)</span></code></pre></div>
<pre><code><span class="co">#&gt; [1] 8860   50</span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># This is just to differentiate ATAC/RNA from the combined matrix of CCA components so they can be separated for pairing</span>
<span class="co"># Based on barcode naming conventions to differentiate ATAC/RNA in our</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">CCA_PCs</span><span class="op">)</span><span class="op">)</span> <span class="co"># ATAC cells</span></code></pre></div>
<pre><code><span class="co">#&gt; [1] "Control_1h_Donor1_S1_BC0004_N01" "Control_1h_Donor1_S1_BC0006_N01"</span>
<span class="co">#&gt; [3] "Control_1h_Donor1_S1_BC0008_N01" "Control_1h_Donor1_S1_BC0009_N01"</span>
<span class="co">#&gt; [5] "Control_1h_Donor1_S1_BC0010_N01" "Control_1h_Donor1_S1_BC0012_N01"</span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">CCA_PCs</span><span class="op">)</span><span class="op">)</span> <span class="co"># RNA cells</span></code></pre></div>
<pre><code><span class="co">#&gt; [1] "Control_1h_Donor4_tgtagtggagttgcacgttgg"</span>
<span class="co">#&gt; [2] "Control_1h_Donor4_aatggccgcacagccgcgctt"</span>
<span class="co">#&gt; [3] "Control_1h_Donor4_cggccaggtcggttttggtta"</span>
<span class="co">#&gt; [4] "Control_1h_Donor4_ggcaggctccttaactggcat"</span>
<span class="co">#&gt; [5] "Control_1h_Donor4_gcagtgtactacgactggcat"</span>
<span class="co">#&gt; [6] "Control_1h_Donor4_tcagcaatcgcgcattcctct"</span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">isATAC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"BC"</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">CCA_PCs</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">isATAC</span><span class="op">)</span> <span class="co"># ATAC vs RNA</span></code></pre></div>
<pre><code><span class="co">#&gt; isATAC</span>
<span class="co">#&gt; FALSE  TRUE </span>
<span class="co">#&gt;  3508  5352</span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ATACcells</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">CCA_PCs</span><span class="op">)</span><span class="op">[</span><span class="va">isATAC</span><span class="op">]</span>
<span class="va">RNAcells</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">CCA_PCs</span><span class="op">)</span><span class="op">[</span><span class="op">!</span><span class="va">isATAC</span><span class="op">]</span>

<span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"ATAC cells in condition: "</span>,<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ATACcells</span><span class="op">)</span>,<span class="st">"\n"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; ATAC cells in condition:  5352</span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"RNA cells in condition: "</span>,<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">RNAcells</span><span class="op">)</span>,<span class="st">"\n"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; RNA cells in condition:  3508</span></code></pre>
<p>We can now visualize UMAP of the ATAC-RNA cell co-embedding, coloring cells by the assay they come from (similar / shared cell states across the two assays should co-localize to some degree in 2D)</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">nPCs</span> <span class="op">&lt;-</span> <span class="fl">20</span> <span class="co"># Num CCA PCs to use when running UMAP / pairing</span>

<span class="co"># Run UMAP</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>
<span class="va">umap.out</span> <span class="op">&lt;-</span> <span class="fu">uwot</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/uwot/man/umap.html" class="external-link">umap</a></span><span class="op">(</span><span class="va">CCA_PCs</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="va">nPCs</span><span class="op">]</span>,
                       metric<span class="op">=</span><span class="st">"cosine"</span>,
                       n_neighbors<span class="op">=</span><span class="fl">30</span><span class="op">)</span>

<span class="va">umap.d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">umap.out</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">umap.d</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UMAP1"</span>,<span class="st">"UMAP2"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">umap.d</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">CCA_PCs</span><span class="op">)</span>
  
<span class="va">umap.d</span><span class="op">$</span><span class="va">Assay</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">isATAC</span>,<span class="st">"ATAC"</span>,<span class="st">"RNA"</span><span class="op">)</span>

<span class="fu">BuenColors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BuenColors/man/shuf.html" class="external-link">shuf</a></span><span class="op">(</span><span class="va">umap.d</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">UMAP1</span>,<span class="va">UMAP2</span>,color<span class="op">=</span><span class="va">Assay</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cadetblue"</span>,<span class="st">"darkorange"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span>override.aes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="FigR_stim_files/figure-html/CCAumap-1.png" width="576" style="display: block; margin: auto;"></p>
<p>Now, we use these co-embeddings to pair cells using our geodesic distance-based pairing approach. We do this step using the <code>pairCells</code> function, providing the separate ATAC and RNA cell components as input as highlighted below:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ATAC_PCs</span> <span class="op">&lt;-</span> <span class="va">CCA_PCs</span><span class="op">[</span><span class="va">isATAC</span>,<span class="op">]</span>
<span class="va">RNA_PCs</span> <span class="op">&lt;-</span> <span class="va">CCA_PCs</span><span class="op">[</span><span class="op">!</span><span class="va">isATAC</span>,<span class="op">]</span>

<span class="va">pairing</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pairCells.html">pairCells</a></span><span class="op">(</span>ATAC <span class="op">=</span> <span class="va">ATAC_PCs</span>,
                     RNA <span class="op">=</span> <span class="va">RNA_PCs</span>,
                     keepUnique <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>By setting the <code>keepUnique</code> parameter specified above to <code>TRUE</code>, we further filter cells such that each ATAC cell barcode only appears once in the resulting pairs (in the case above).</p>
<p>We can now visualize the resulting ATAC-RNA pairs on the CCA UMAP</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotPairs.html">plotPairs</a></span><span class="op">(</span>ATAC <span class="op">=</span> <span class="va">pairing</span><span class="op">$</span><span class="va">ATAC</span>,
          RNA<span class="op">=</span><span class="va">pairing</span><span class="op">$</span><span class="va">RNA</span>,
          max.show <span class="op">=</span> <span class="fl">100</span>,
          umap.df <span class="op">=</span> <span class="va">umap.d</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; Assuming concordance in pairs ..</span>
<span class="co">#&gt; Only highlighting 100 pairs at random..</span></code></pre>
<pre><code><span class="co">#&gt; Loading required package: ggrastr</span></code></pre>
<pre><code><span class="co">#&gt; Warning: package 'ggrastr' was built under R version 4.0.4</span></code></pre>
<p><img src="FigR_stim_files/figure-html/plotPairs-1.png" width="576" style="display: block; margin: auto;"></p>
<p>As the final step, we subset the ATAC and RNA count objects to match the determined ATAC-RNA paired cells, and use these as input in later steps</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ATAC.se.paired</span> <span class="op">&lt;-</span> <span class="va">ATAC.se</span><span class="op">[</span>,<span class="va">pairing</span><span class="op">$</span><span class="va">ATAC</span><span class="op">]</span>
<span class="va">RNAmat.paired</span> <span class="op">&lt;-</span> <span class="va">RNAmat</span><span class="op">[</span>,<span class="va">pairing</span><span class="op">$</span><span class="va">RNA</span><span class="op">]</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="peak-gene-association-testing">Peak-gene association testing<a class="anchor" aria-label="anchor" href="#peak-gene-association-testing"></a>
</h3>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Don't run interactively</span>
<span class="va">cisCorr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runGenePeakcorr.html">runGenePeakcorr</a></span><span class="op">(</span>ATAC.se <span class="op">=</span> <span class="va">ATAC.se.paired</span>,
                           RNAmat <span class="op">=</span> <span class="va">RNAmat.paired</span>,
                           genome <span class="op">=</span> <span class="st">"hg19"</span>, <span class="co"># One of hg19, mm10 or hg38 </span>
                           nCores <span class="op">=</span> <span class="fl">4</span>,
                           p.cut <span class="op">=</span> <span class="cn">NULL</span>, <span class="co"># Set this to NULL and we can filter later</span>
                           n_bg <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="determining-dorcs">Determining DORCs<a class="anchor" aria-label="anchor" href="#determining-dorcs"></a>
</h3>
<p>We can now filter correlations based on the background p-value, and summarize the genes that have a relatively high number of peak-gene associations (i.e. Domains of regulatory chromatin or ‘DORCs’)</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cisCorr</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt;   Peak         PeakRanges      Gene        rObs       pvalZ</span>
<span class="co">#&gt; 1    2 chr1:713966-714266 LINC01128 0.018159834 0.100025782</span>
<span class="co">#&gt; 2    5 chr1:756685-756985 LINC01128 0.004361716 0.318375663</span>
<span class="co">#&gt; 3    8 chr1:777265-777565 LINC01128 0.006061255 0.165574826</span>
<span class="co">#&gt; 4   19 chr1:839948-840248    SAMD11 0.003911159 0.289553871</span>
<span class="co">#&gt; 5   23 chr1:848168-848468    SAMD11 0.045598153 0.005385101</span>
<span class="co">#&gt; 6   26 chr1:859076-859376    SAMD11 0.031427158 0.107115444</span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cisCorr.filt</span> <span class="op">&lt;-</span> <span class="va">cisCorr</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pvalZ</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="va">dorcGenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dorcJPlot.html">dorcJPlot</a></span><span class="op">(</span>dorcTab <span class="op">=</span> <span class="va">cisCorr.filt</span>,
                         cutoff <span class="op">=</span> <span class="fl">10</span>, <span class="co"># No. sig peaks needed to be called a DORC</span>
                         labelTop <span class="op">=</span> <span class="fl">20</span>,
                         returnGeneList <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Set this to FALSE for just the plot</span>
                         force<span class="op">=</span><span class="fl">5</span><span class="op">)</span></code></pre></div>
<p><img src="FigR_stim_files/figure-html/getDORCs-1.png" width="576" style="display: block; margin: auto;"></p>
<p>We can also just get the full ranked table of genes and number of significantly associated peaks, instead of only getting the gene names (so you can threshold yourself)</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Unfiltered</span>
<span class="va">numDorcs</span> <span class="op">&lt;-</span> <span class="va">cisCorr.filt</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Gene</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">tally</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span>
<span class="va">numDorcs</span></code></pre></div>
<pre><code><span class="co">#&gt; <span style="color: #949494;"># A tibble: 9,272 x 2</span></span>
<span class="co">#&gt;    Gene         n</span>
<span class="co">#&gt;    <span style="color: #949494;font-style: italic;">&lt;chr&gt;</span><span>    </span><span style="color: #949494;font-style: italic;">&lt;int&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span><span> IL7R        23</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span><span> TCF7        23</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span><span> CD83        22</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span><span> SEMA7A      20</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span><span> APOBEC3G    19</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span><span> CCR7        18</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span><span> CD8A        18</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span><span> PAX5        18</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span><span> SLC7A5      18</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span><span> ANPEP       17</span></span>
<span class="co">#&gt; <span style="color: #949494;"># ... with 9,262 more rows</span></span></code></pre>
<p>Since these are unstimulated (control) PBMCs, we expect most of these genes to be lineage-determining markers.</p>
</div>
<div class="section level3">
<h3 id="visualizing-dorcs">Visualizing DORCs<a class="anchor" aria-label="anchor" href="#visualizing-dorcs"></a>
</h3>
<p>To get the DORC accessibility scores, we can sum up the chromatin accessibility peak counts for peaks associated with a given gene.</p>
<blockquote>
<blockquote>
<p>NOTE: It is crucial that you use the exact same <code>SummarizedExperiment</code> object here as the one that was used in the <code>runGenePeakcorr</code> step</p>
</blockquote>
</blockquote>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dorcMat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDORCScores.html">getDORCScores</a></span><span class="op">(</span>ATAC.se <span class="op">=</span> <span class="va">ATAC.se.paired</span>, <span class="co"># Has to be same SE as used in previous step</span>
                         dorcTab <span class="op">=</span> <span class="va">cisCorr.filt</span>,
                         geneList <span class="op">=</span> <span class="va">dorcGenes</span>,
                         nCores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; ........</span>
<span class="co">#&gt; Normalizing scATAC counts ..</span>
<span class="co">#&gt; Centering counts for cells sequentially in groups of size  5000  ..</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Computing centered counts for cells:  1  to  4912 ..</span>
<span class="co">#&gt; Computing centered counts per cell using mean reads in features ..</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Merging results..</span>
<span class="co">#&gt; Done!</span>
<span class="co">#&gt; Computing DORC scores ..</span>
<span class="co">#&gt; Running in parallel using  1 cores ..</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Time Elapsed:  10.3236038684845 secs</span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">dorcMat</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; [1]   99 4912</span></code></pre>
<p>We then plot these scores, along with the paired RNA expression onto the original scATAC-seq UMAP. Here’s a look at what that 2D embedding looks like (also highlighted in the manuscript), just for this subset of cells:</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">ATAC.se.paired</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">UMAP1</span>,<span class="va">UMAP2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"gray"</span>,size<span class="op">=</span><span class="fl">0.8</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="FigR_stim_files/figure-html/plotATAC-1.png" width="576" style="display: block; margin: auto;"></p>
<p>We can then smooth these (sparse) DORC counts, and visualize side-by-side with the expression, on the same UMAP. In this particular case, we use the original Latent Semantic Indexing (LSI) cell embeddings that were pre-determined using all cells (using which the 2D UMAP was determined), subset to the cells being used, and then derive cell k-nearest neighbors, which we will use to smooth both the DORC and [aired RNA expression, for the same UMAP:</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">lsi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"./control1h_PBMC_atac_lsi.rds"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">lsi</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">dorcMat</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">lsi</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Subset to paired ATAC</span>
<span class="va">lsi</span> <span class="op">&lt;-</span> <span class="va">lsi</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">dorcMat</span><span class="op">)</span>,<span class="op">]</span>

<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">lsi</span><span class="op">)</span>

<span class="co"># Get cell KNNs</span>
<span class="va">cellkNN</span> <span class="op">&lt;-</span> <span class="fu">FNN</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/FNN/man/get.knn.html" class="external-link">get.knn</a></span><span class="op">(</span><span class="va">lsi</span>,k<span class="op">=</span><span class="fl">30</span><span class="op">)</span><span class="op">$</span><span class="va">nn.index</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">cellkNN</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">dorcMat</span><span class="op">)</span>

<span class="co"># Smooth dorc scores using cell KNNs (k=30)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">doParallel</span><span class="op">)</span>
<span class="va">dorcMat.s</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/smoothScoresNN.html">smoothScoresNN</a></span><span class="op">(</span>NNmat <span class="op">=</span> <span class="va">cellkNN</span>,mat <span class="op">=</span> <span class="va">dorcMat</span>,nCores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>

<span class="co"># Smooth RNA using cell KNNs</span>
<span class="co"># This takes longer since it's all genes</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">RNAmat.paired</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">ATAC.se.paired</span><span class="op">)</span> <span class="co"># Just so that the smoothing function doesn't throw an error (matching cell barcodes in the KNN and the matrix)</span>
<span class="va">RNAmat.s</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/smoothScoresNN.html">smoothScoresNN</a></span><span class="op">(</span>NNmat <span class="op">=</span> <span class="va">cellkNN</span>,mat <span class="op">=</span> <span class="va">RNAmat.paired</span>,nCores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Visualize on pre-computed UMAP</span>
<span class="co"># This is the ATAC UMAP shown in the paper (based on ATAC LSI)</span>
<span class="va">umap.d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">ATAC.se.paired</span><span class="op">)</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UMAP1"</span>,<span class="st">"UMAP2"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>

<span class="co"># DORC scores for top DORC(s)</span>
<span class="va">myDORCs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SEMA7A"</span>,<span class="st">"IL7R"</span>,<span class="st">"CD93"</span><span class="op">)</span>

<span class="va">dorcGGlist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">myDORCs</span>,<span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span> 
  <span class="fu"><a href="../reference/plotMarker2D.html">plotMarker2D</a></span><span class="op">(</span><span class="va">umap.d</span>,
               <span class="va">dorcMat.s</span>,
               markers <span class="op">=</span> <span class="va">x</span>,
               maxCutoff <span class="op">=</span> <span class="st">"q0.99"</span>,
               colorPalette <span class="op">=</span> <span class="st">"brewer_heat"</span>
  <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">x</span>,<span class="st">" DORC"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; Plotting  SEMA7A </span>
<span class="co">#&gt; Plotting  IL7R </span>
<span class="co">#&gt; Plotting  CD93</span></code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Paired RNA expression for top DORC(s)</span>
<span class="co"># Plot on the same reference ATAC UMAP</span>
<span class="va">rnaGGlist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">myDORCs</span>,<span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span> 
  <span class="fu"><a href="../reference/plotMarker2D.html">plotMarker2D</a></span><span class="op">(</span><span class="va">umap.d</span>,
               <span class="va">RNAmat.s</span>,
               markers <span class="op">=</span> <span class="va">x</span>,
               maxCutoff <span class="op">=</span> <span class="st">"q0.99"</span>,
               colorPalette <span class="op">=</span> <span class="st">"brewer_purple"</span>
  <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">x</span>,<span class="st">" RNA"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; Plotting  SEMA7A </span>
<span class="co">#&gt; Plotting  IL7R </span>
<span class="co">#&gt; Plotting  CD93</span></code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; Warning: package 'patchwork' was built under R version 4.0.4</span></code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">dorcGGlist</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="va">dorcGGlist</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="va">dorcGGlist</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">/</span>  <span class="op">(</span><span class="va">rnaGGlist</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="va">rnaGGlist</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="va">rnaGGlist</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="FigR_stim_files/figure-html/visDORCs-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="tf-gene-associations">TF-gene associations<a class="anchor" aria-label="anchor" href="#tf-gene-associations"></a>
</h3>
<p>As before, we can now determine TF-gene associations, and begin inferring a regulatory network based on the previous DORC definitions, together with information drawn from a database of TF binding sequence motifs</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">figR.d</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runFigRGRN.html">runFigRGRN</a></span><span class="op">(</span>ATAC.se <span class="op">=</span> <span class="va">ATAC.se.paired</span>, <span class="co"># Must be the same input as used in runGenePeakcorr()</span>
                     dorcTab <span class="op">=</span> <span class="va">cisCorr.filt</span>, <span class="co"># Filtered peak-gene associations</span>
                     genome <span class="op">=</span> <span class="st">"hg19"</span>,
                     dorcMat <span class="op">=</span> <span class="va">dorcMat.s</span>,
                     dorcK <span class="op">=</span> <span class="fl">5</span>, 
                     rnaMat <span class="op">=</span> <span class="va">RNAmat.s</span>, 
                     nCores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visualizing-figr-results">Visualizing FigR results<a class="anchor" aria-label="anchor" href="#visualizing-figr-results"></a>
</h3>
<div class="section level4">
<h4 id="global-regulation-profile">Global regulation profile<a class="anchor" aria-label="anchor" href="#global-regulation-profile"></a>
</h4>
<p>Scatter of correlation (x) and motif enrichment (y)</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">figR.d</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">Corr.log10P</span>,<span class="va">Enrichment.log10P</span>,color<span class="op">=</span><span class="va">Score</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">ggrastr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggrastr/man/geom_point_rast.html" class="external-link">geom_point_rast</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">0.01</span>,shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_color_gradientn</a></span><span class="op">(</span>colours <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BuenColors/man/jdb_palette.html" class="external-link">jdb_palette</a></span><span class="op">(</span><span class="st">"solar_extra"</span><span class="op">)</span>,limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>,<span class="fl">3</span><span class="op">)</span>,oob <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org/reference/oob.html" class="external-link">squish</a></span>,breaks<span class="op">=</span><span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org/reference/breaks_pretty.html" class="external-link">breaks_pretty</a></span><span class="op">(</span>n<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="FigR_stim_files/figure-html/plotallFigR-1.png" width="480" style="display: block; margin: auto;"></p>
<p>Looking at the top TFs associated with DORCs</p>
</div>
<div class="section level4">
<h4 id="ranking-tf-drivers">Ranking TF drivers<a class="anchor" aria-label="anchor" href="#ranking-tf-drivers"></a>
</h4>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/rankDrivers.html">rankDrivers</a></span><span class="op">(</span><span class="va">figR.d</span>,rankBy <span class="op">=</span> <span class="st">"meanScore"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; Warning: ggrepel: 63 unlabeled data points (too many overlaps). Consider</span>
<span class="co">#&gt; increasing max.overlaps</span></code></pre>
<p><img src="FigR_stim_files/figure-html/rankDrivers-1.png" width="700"></p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/rankDrivers.html">rankDrivers</a></span><span class="op">(</span><span class="va">figR.d</span>,score.cut <span class="op">=</span> <span class="fl">1.5</span>,rankBy <span class="op">=</span> <span class="st">"nTargets"</span>,interactive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-775d2ff30998a5f26368" style="width:700px;height:384px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-775d2ff30998a5f26368">{"x":{"data":[{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9,0.9,0.9,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.900000000000002,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006],"base":[0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,-2,-2,0,0,0,-11,0,0,0,0,0,0,0,0],"x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40],"y":[22,19,19,19,20,14,6,5,5,4,4,3,3,3,2,2,2,1,1,1,1,1,1,1,1,1,1,1,0,0,0,5,0,0,0,0,0,0,0,0],"text":["Motif: SPI1<br />numActivated: 22<br />numRepressed: 0","Motif: CEBPA<br />numActivated: 19<br />numRepressed: 0","Motif: CEBPB<br />numActivated: 19<br />numRepressed: 0","Motif: FOSL1<br />numActivated: 19<br />numRepressed: 0","Motif: MAFG<br />numActivated: 20<br />numRepressed: 1","Motif: JUNB<br />numActivated: 14<br />numRepressed: 0","Motif: MYBL1<br />numActivated:  6<br />numRepressed: 0","Motif: FOSL2<br />numActivated:  5<br />numRepressed: 0","Motif: POU2F2<br />numActivated:  5<br />numRepressed: 0","Motif: RUNX3<br />numActivated:  4<br />numRepressed: 0","Motif: VDR<br />numActivated:  4<br />numRepressed: 0","Motif: CEBPD<br />numActivated:  3<br />numRepressed: 0","Motif: IRF4<br />numActivated:  3<br />numRepressed: 0","Motif: RUNX1<br />numActivated:  3<br />numRepressed: 0","Motif: EOMES<br />numActivated:  2<br />numRepressed: 0","Motif: IRF8<br />numActivated:  2<br />numRepressed: 0","Motif: RXRA<br />numActivated:  2<br />numRepressed: 0","Motif: BATF<br />numActivated:  1<br />numRepressed: 0","Motif: GTF3A<br />numActivated:  1<br />numRepressed: 0","Motif: MEF2C<br />numActivated:  1<br />numRepressed: 0","Motif: MSC<br />numActivated:  1<br />numRepressed: 0","Motif: NR4A2<br />numActivated:  1<br />numRepressed: 0","Motif: PLAGL1<br />numActivated:  1<br />numRepressed: 0","Motif: RORA<br />numActivated:  1<br />numRepressed: 0","Motif: SPIB<br />numActivated:  1<br />numRepressed: 0","Motif: TCF3<br />numActivated:  1<br />numRepressed: 0","Motif: LYL1<br />numActivated:  1<br />numRepressed: 1","Motif: ZEB2<br />numActivated:  1<br />numRepressed: 1","Motif: CXXC5<br />numActivated:  0<br />numRepressed: 1","Motif: FOXO1<br />numActivated:  0<br />numRepressed: 1","Motif: KLF2<br />numActivated:  0<br />numRepressed: 1","Motif: MAFB<br />numActivated:  5<br />numRepressed: 6","Motif: MEF2A<br />numActivated:  0<br />numRepressed: 1","Motif: NFATC2<br />numActivated:  0<br />numRepressed: 1","Motif: OLIG1<br />numActivated:  0<br />numRepressed: 1","Motif: BACH2<br />numActivated:  0<br />numRepressed: 2","Motif: BCL11B<br />numActivated:  0<br />numRepressed: 5","Motif: HMGA1<br />numActivated:  0<br />numRepressed: 6","Motif: ZBTB18<br />numActivated:  0<br />numRepressed: 6","Motif: MAFF<br />numActivated:  0<br />numRepressed: 8"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(205,38,38,1)","line":{"width":0.377952755905512,"color":"rgba(211,211,211,1)"}},"name":"numActivatedY","legendgroup":"numActivatedY","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9,0.9,0.9,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.900000000000002,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006,0.900000000000006],"base":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,-1,-1,-1,-1,-1,-6,-1,-1,-1,-2,-5,-6,-6,-8],"x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40],"y":[0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,1,1,6,1,1,1,2,5,6,6,8],"text":["Motif: SPI1<br />numActivated: 22<br />numRepressed: 0","Motif: CEBPA<br />numActivated: 19<br />numRepressed: 0","Motif: CEBPB<br />numActivated: 19<br />numRepressed: 0","Motif: FOSL1<br />numActivated: 19<br />numRepressed: 0","Motif: MAFG<br />numActivated: 20<br />numRepressed: 1","Motif: JUNB<br />numActivated: 14<br />numRepressed: 0","Motif: MYBL1<br />numActivated:  6<br />numRepressed: 0","Motif: FOSL2<br />numActivated:  5<br />numRepressed: 0","Motif: POU2F2<br />numActivated:  5<br />numRepressed: 0","Motif: RUNX3<br />numActivated:  4<br />numRepressed: 0","Motif: VDR<br />numActivated:  4<br />numRepressed: 0","Motif: CEBPD<br />numActivated:  3<br />numRepressed: 0","Motif: IRF4<br />numActivated:  3<br />numRepressed: 0","Motif: RUNX1<br />numActivated:  3<br />numRepressed: 0","Motif: EOMES<br />numActivated:  2<br />numRepressed: 0","Motif: IRF8<br />numActivated:  2<br />numRepressed: 0","Motif: RXRA<br />numActivated:  2<br />numRepressed: 0","Motif: BATF<br />numActivated:  1<br />numRepressed: 0","Motif: GTF3A<br />numActivated:  1<br />numRepressed: 0","Motif: MEF2C<br />numActivated:  1<br />numRepressed: 0","Motif: MSC<br />numActivated:  1<br />numRepressed: 0","Motif: NR4A2<br />numActivated:  1<br />numRepressed: 0","Motif: PLAGL1<br />numActivated:  1<br />numRepressed: 0","Motif: RORA<br />numActivated:  1<br />numRepressed: 0","Motif: SPIB<br />numActivated:  1<br />numRepressed: 0","Motif: TCF3<br />numActivated:  1<br />numRepressed: 0","Motif: LYL1<br />numActivated:  1<br />numRepressed: 1","Motif: ZEB2<br />numActivated:  1<br />numRepressed: 1","Motif: CXXC5<br />numActivated:  0<br />numRepressed: 1","Motif: FOXO1<br />numActivated:  0<br />numRepressed: 1","Motif: KLF2<br />numActivated:  0<br />numRepressed: 1","Motif: MAFB<br />numActivated:  5<br />numRepressed: 6","Motif: MEF2A<br />numActivated:  0<br />numRepressed: 1","Motif: NFATC2<br />numActivated:  0<br />numRepressed: 1","Motif: OLIG1<br />numActivated:  0<br />numRepressed: 1","Motif: BACH2<br />numActivated:  0<br />numRepressed: 2","Motif: BCL11B<br />numActivated:  0<br />numRepressed: 5","Motif: HMGA1<br />numActivated:  0<br />numRepressed: 6","Motif: ZBTB18<br />numActivated:  0<br />numRepressed: 6","Motif: MAFF<br />numActivated:  0<br />numRepressed: 8"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(54,100,139,1)","line":{"width":0.377952755905512,"color":"rgba(211,211,211,1)"}},"name":"numRepressedY","legendgroup":"numRepressedY","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.4,40.6],"y":[0,0],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.3181126331811,"r":7.30593607305936,"b":31.5829528158295,"l":37.2602739726027},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.4,40.6],"tickmode":"array","ticktext":["SPI1","CEBPA","CEBPB","FOSL1","MAFG","JUNB","MYBL1","FOSL2","POU2F2","RUNX3","VDR","CEBPD","IRF4","RUNX1","EOMES","IRF8","RXRA","BATF","GTF3A","MEF2C","MSC","NR4A2","PLAGL1","RORA","SPIB","TCF3","LYL1","ZEB2","CXXC5","FOXO1","KLF2","MAFB","MEF2A","NFATC2","OLIG1","BACH2","BCL11B","HMGA1","ZBTB18","MAFF"],"tickvals":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40],"categoryorder":"array","categoryarray":["SPI1","CEBPA","CEBPB","FOSL1","MAFG","JUNB","MYBL1","FOSL2","POU2F2","RUNX3","VDR","CEBPD","IRF4","RUNX1","EOMES","IRF8","RXRA","BATF","GTF3A","MEF2C","MSC","NR4A2","PLAGL1","RORA","SPIB","TCF3","LYL1","ZEB2","CXXC5","FOXO1","KLF2","MAFB","MEF2A","NFATC2","OLIG1","BACH2","BCL11B","HMGA1","ZBTB18","MAFF"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":{"text":"Ranked TF Motifs","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-12.65,23.65],"tickmode":"array","ticktext":[10,0,10,20],"tickvals":[-10,0,10,20],"categoryorder":"array","categoryarray":[10,0,10,20],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":{"text":"# Associated genes <br />abs(Score) >= 1.5","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"4f8c165f7cf3":{"x":{},"y":{},"fill":{},"numActivated":{},"numRepressed":{},"type":"bar"},"4f8c4f03184":{"yintercept":{}}},"cur_data":"4f8c165f7cf3","visdat":{"4f8c165f7cf3":["function (y) ","x"],"4f8c4f03184":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script><p>You can also set the <code>interactive</code> parameter (which the rankDrivers function takes) to <code>TRUE</code>, which will let you hover over points and provide some extra information in terms of the number of estimated activated vs repressed genes</p>
</div>
<div class="section level4">
<h4 id="heatmap-view">Heatmap view<a class="anchor" aria-label="anchor" href="#heatmap-view"></a>
</h4>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span>
<span class="fu"><a href="../reference/plotfigRHeatmap.html">plotfigRHeatmap</a></span><span class="op">(</span>figR.d <span class="op">=</span> <span class="va">figR.d</span>,
                score.cut <span class="op">=</span> <span class="fl">1.5</span>,
                column_names_gp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/gpar.html" class="external-link">gpar</a></span><span class="op">(</span>fontsize<span class="op">=</span><span class="fl">6</span><span class="op">)</span>, <span class="co"># from ComplexHeatmap</span>
                show_row_dend <span class="op">=</span> <span class="cn">FALSE</span> <span class="co"># from ComplexHeatmap</span>
                <span class="op">)</span></code></pre></div>
<p><img src="FigR_stim_files/figure-html/plotHeat-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="network-view">Network view<a class="anchor" aria-label="anchor" href="#network-view"></a>
</h4>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://CRAN.R-project.org/package=networkD3" class="external-link">networkD3</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">#&gt; Warning: package 'networkD3' was built under R version 4.0.5</span></code></pre>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotfigRNetwork.html">plotfigRNetwork</a></span><span class="op">(</span><span class="va">figR.d</span>,
                      score.cut <span class="op">=</span> <span class="fl">1.5</span>,
                      weight.edges <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-1883ceb31574f017a62e" style="width:700px;height:432.632880098888px;" class="forceNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-1883ceb31574f017a62e">{"x":{"links":{"source":[0,0,1,1,2,2,2,2,3,3,3,3,4,5,5,6,6,7,8,8,9,9,9,10,10,10,10,10,11,11,11,12,13,13,13,13,13,13,13,14,15,16,17,17,17,18,18,18,18,18,18,18,19,20,20,20,20,21,21,22,22,22,23,24,24,24,24,25,25,25,25,25,25,25,26,27,28,28,28,29,29,29,29,30,31,31,32,32,32,32,32,33,33,33,33,33,33,33,34,35,36,36,37,37,37,37,38,38,38,38,39,40,40,40,41,41,42,42,42,42,42,42,42,43,43,44,45,45,45,45,45,45,45,46,46,47,47,47,47,47,47,47,47,47,47,48,49,50,50,51,51,52,53,53,53,53,53,53,54,54,55,56,56,56,56,56,56,56,56,57,57,57,57,57,58,58,59,59,59,60,61,61,61,62,62,62,62,62,62,63,63,64,64,65,66,66,66,66,66,66,66,67,68,68,68,68,68,68,68,68],"target":[69,70,71,72,69,73,74,75,69,75,76,70,77,78,79,80,76,78,81,72,77,82,72,75,83,84,85,86,85,87,88,89,69,73,75,85,70,86,90,86,78,91,91,70,92,69,73,75,83,85,70,86,82,69,73,74,93,87,72,94,82,72,77,69,73,70,86,69,73,75,83,85,70,86,86,71,69,74,95,75,85,70,86,76,78,96,69,73,75,93,86,69,73,75,85,70,86,90,78,71,76,70,94,82,72,79,73,70,97,86,98,75,70,86,99,71,100,69,73,75,85,70,86,80,76,78,69,73,75,76,70,86,90,80,76,101,100,73,75,83,85,76,99,95,86,102,78,69,73,93,103,104,69,73,75,85,93,86,70,86,76,100,69,73,75,85,76,70,86,100,75,85,70,86,105,82,69,73,93,106,76,70,107,100,69,73,75,70,86,77,82,77,108,77,69,73,75,85,70,86,90,78,101,73,75,83,85,70,99,86],"value":[15.7794721678651,16.549022422104,15.7911444514124,1.88710057760428,15.8688569202072,16.6805354609319,15.6973712135359,16.0167722482471,15.7532536564056,17.2409044034944,16.3464790968575,17.3378018099466,3.92035806624016,3.91524370442152,3.30051514596731,16.5765935369459,3.83269023954571,4.27887906645658,3.36922979926178,3.86100753864929,3.26149382112455,19.5395309975922,3.35552307841404,16.8444795643509,17.0153038337136,3.97983965163949,16.8764028052398,17.3125314816739,16.6995379877404,17.1436406794181,4.22310937111966,3.24679817824003,15.4350025453998,16.592134739525,17.1078489916205,15.5735460424001,17.0552543491413,19.639011196778,15.6112660470293,15.7824432372991,3.69948918172611,15.8506604865021,15.7050488281144,15.981289119742,15.6918788567718,15.5616044901883,16.2677360329673,17.6588667012221,15.1003619516171,16.0298879369621,18.0974184385038,19.0012743117717,17.5508832992771,15.7817953111639,17.2022701096264,15.5521522938125,14.993332373658,2.6852205253584,4.31076756626025,16.6813260937102,16.6737872347192,4.18608369677906,3.47798079892911,15.2373464238147,16.4719779820683,16.7942473775024,17.085487777002,16.0954664537711,16.5075531541699,17.1002942412934,17.2938991261678,15.234630579028,15.1180237930257,17.422399864326,17.2905655031187,15.7041759437303,15.3543458898595,15.23794570927,15.3942085300283,15.3031167851412,16.9308263494985,17.177827062501,20,3.73204478299499,4.33076682283332,4.13077490364409,15.665053591637,17.2927455898436,16.2316563915671,16.0600945663794,18.7895995845117,16.0544037502275,17.0756336741461,16.9544578242904,15.491511213737,17.3632195857548,19.7815865906325,15.6859074801137,4.10389627706548,15.9894269482919,2.48803555050168,3.50992428003369,16.6564918596212,16.280765329689,2.04673675660678,15.3770782217703,15.8309094669219,16.9949709340166,16.398375011826,17.1575749872281,15.1999570703777,16.0756412904984,17.571738686069,19.5886029085629,15.622674465118,15.5218547502265,3.94131046550118,15.2415141035917,15.4395909045922,17.2617039522031,16.7464850486433,17.0378818083392,19.2780903590883,16.4668344111137,3.87684543733438,3.98669426236579,15.7233641723789,17.4311729701097,16.8477499810252,15.2795778508781,16.9212222782926,17.1249196541721,18.4416362989913,15.8039222285229,3.95519407523306,3.81400541025608,3.78855550362632,15.0228583910931,17.0433194458559,16.1994394230132,16.9124110161518,15.8413842712926,15.6638386181376,16.0840751660741,18.313044134126,15.8007781774692,3.92528235907826,15.9790165210063,17.3600367214456,15.8648662004363,15.3030189120036,15.059666228249,16.4928803769393,18.5873856011174,17.0922293122755,15.5445213104853,15.9305412507836,19.2756524701799,18.4330993068121,16.8402019364043,4.20150220294796,3.58725774549142,15.0482192777201,16.8230093987227,17.8994587789062,17.36010845258,15.3056181807644,16.4041646883556,19.0238135654375,4.08509849429303,16.9634360530429,15.5490406420557,16.3449904403932,18.1011918194308,17.5325274294545,17.8272690356679,15.7656464721041,16.1716932075572,15.3778102739754,3.31561847755723,16.5366798051955,17.7776347967149,15.1232978668462,4.33262160794811,15.0088919414456,15.8749184359553,16.7828046062988,15.9440331038968,17.3119882769944,0,15.3323064884295,2.32650311760628,14.993961497784,4.17812068632265,15.1777141764779,16.9554702590612,17.322086487509,15.5648980989449,17.4660784325237,19.6498091841491,15.6290407854215,4.25861110529176,3.73088815309572,15.6066965021821,18.4689396967679,15.1156444793076,16.3008323277317,15.0627012916929,15.4615324121344,19.1580769615133],"colour":["#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#A020F0","#A020F0","#228B22","#A020F0","#A020F0","#A020F0","#A020F0","#A020F0","#228B22","#A020F0","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#A020F0","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#A020F0","#228B22","#228B22","#A020F0","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#A020F0","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#A020F0","#A020F0","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#A020F0","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#228B22","#A020F0","#228B22","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#A020F0","#A020F0","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22","#228B22"]},"nodes":{"name":["ADAP1","ADGRG1","AGPAT2","ANPEP","APOBEC3G","ARL4C","BAIAP3","BCL11B","CCL4","CCL5","CCR2","CD22","CD28","CD300C","CD300E","CD5","CD79B","CD83","CD93","CHST12","CSF1R","CST7","CX3CR1","DUSP2","ENG","FAM129B","FCAR","GZMB","HMOX1","IRAK2","IRF4","ITK","LILRB2","LINC01272","MAL","METRNL","MGAT5","NKG7","NOTCH1","NUAK2","OLR1","OSBPL5","OSM","PAX5","PIK3IP1","PLAUR","POU2AF1","PPIF","PRF1","PRR12","QSOX1","REL","RGS10","RXRA","SEMA6B","SEMA7A","SH3BP2","SLC43A2","SLC7A5","SPI1","SPON2","SRC","TBC1D22A","TBKBP1","TBX21","THEMIS","TNS1","TRAT1","VEGFA","CEBPA.","MAFG.","RUNX3.","ZBTB18.","CEBPB.","CEBPD.","FOSL1.","MAFB.","HMGA1.","MAFF.","ZEB2.","IRF4.","KLF2.","MYBL1.","FOSL2.","FOXO1.","JUNB.","SPI1.","LYL1.","NFATC2.","CXXC5.","VDR.","IRF8.","PLAGL1.","POU2F2.","EOMES.","RXRA.","OLIG1.","MSC.","TCF3.","RUNX1.","BCL11B.","BACH2.","RORA.","SPIB.","GTF3A.","BATF.","MEF2A.","MEF2C.","NR4A2."],"group":["DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","DORC","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF"],"nodesize":[8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3]},"options":{"NodeID":"name","Group":"group","colourScale":"d3.scaleOrdinal([\" #FF6347\", \"#87CEEB \"])","fontSize":13,"fontFamily":"serif","clickTextSize":32.5,"linkDistance":50,"linkWidth":"function(d) { return Math.sqrt(d.value); }","charge":-15,"opacity":1,"zoom":true,"legend":true,"arrows":false,"nodesize":true,"radiusCalculation":"Math.sqrt(d.nodesize)*2","bounded":true,"opacityNoHover":0.6,"clickAction":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Vinay Kartha.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
